<!doctype html>
<html>
<head>
<meta charset='UTF-8'><meta name='viewport' content='width=device-width initial-scale=1'>

<link href='https://fonts.loli.net/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext' rel='stylesheet' type='text/css' /><style type='text/css'>html {overflow-x: initial !important;}:root { --bg-color: #ffffff; --text-color: #333333; --select-text-bg-color: #B5D6FC; --select-text-font-color: auto; --monospace: "Lucida Console",Consolas,"Courier",monospace; --title-bar-height: 20px; }
.mac-os-11 { --title-bar-height: 28px; }
html { font-size: 14px; background-color: var(--bg-color); color: var(--text-color); font-family: "Helvetica Neue", Helvetica, Arial, sans-serif; -webkit-font-smoothing: antialiased; }
body { margin: 0px; padding: 0px; height: auto; inset: 0px; font-size: 1rem; line-height: 1.42857143; overflow-x: hidden; background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; tab-size: 4; background-position: inherit; background-repeat: inherit; }
iframe { margin: auto; }
a.url { word-break: break-all; }
a:active, a:hover { outline: 0px; }
.in-text-selection, ::selection { text-shadow: none; background: var(--select-text-bg-color); color: var(--select-text-font-color); }
#write { margin: 0px auto; height: auto; width: inherit; word-break: normal; word-wrap: break-word; position: relative; white-space: normal; overflow-x: visible; padding-top: 36px; }
#write.first-line-indent p { text-indent: 2em; }
#write.first-line-indent li p, #write.first-line-indent p * { text-indent: 0px; }
#write.first-line-indent li { margin-left: 2em; }
.for-image #write { padding-left: 8px; padding-right: 8px; }
body.typora-export { padding-left: 30px; padding-right: 30px; }
.typora-export .footnote-line, .typora-export li, .typora-export p { white-space: pre-wrap; }
.typora-export .task-list-item input { pointer-events: none; }
@media screen and (max-width: 500px) { 
  body.typora-export { padding-left: 0px; padding-right: 0px; }
  #write { padding-left: 20px; padding-right: 20px; }
  .CodeMirror-sizer { margin-left: 0px !important; }
  .CodeMirror-gutters { display: none !important; }
}
#write li > figure:last-child { margin-bottom: 0.5rem; }
#write ol, #write ul { position: relative; }
img { max-width: 100%; vertical-align: middle; image-orientation: from-image; }
button, input, select, textarea { color: inherit; font-family: inherit; font-size: inherit; font-style: inherit; font-variant-caps: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; }
input[type="checkbox"], input[type="radio"] { line-height: normal; padding: 0px; }
*, ::after, ::before { box-sizing: border-box; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p, #write pre { width: inherit; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p { position: relative; }
p { line-height: inherit; }
h1, h2, h3, h4, h5, h6 { break-after: avoid-page; break-inside: avoid; orphans: 4; }
p { orphans: 4; }
h1 { font-size: 2rem; }
h2 { font-size: 1.8rem; }
h3 { font-size: 1.6rem; }
h4 { font-size: 1.4rem; }
h5 { font-size: 1.2rem; }
h6 { font-size: 1rem; }
.md-math-block, .md-rawblock, h1, h2, h3, h4, h5, h6, p { margin-top: 1rem; margin-bottom: 1rem; }
.hidden { display: none; }
.md-blockmeta { color: rgb(204, 204, 204); font-weight: 700; font-style: italic; }
a { cursor: pointer; }
sup.md-footnote { padding: 2px 4px; background-color: rgba(238, 238, 238, 0.7); color: rgb(85, 85, 85); border-top-left-radius: 4px; border-top-right-radius: 4px; border-bottom-right-radius: 4px; border-bottom-left-radius: 4px; cursor: pointer; }
sup.md-footnote a, sup.md-footnote a:hover { color: inherit; text-transform: inherit; text-decoration: inherit; }
#write input[type="checkbox"] { cursor: pointer; width: inherit; height: inherit; }
figure { overflow-x: auto; margin: 1.2em 0px; max-width: calc(100% + 16px); padding: 0px; }
figure > table { margin: 0px; }
tr { break-inside: avoid; break-after: auto; }
thead { display: table-header-group; }
table { border-collapse: collapse; border-spacing: 0px; width: 100%; overflow: auto; break-inside: auto; text-align: left; }
table.md-table td { min-width: 32px; }
.CodeMirror-gutters { border-right-width: 0px; background-color: inherit; }
.CodeMirror-linenumber { }
.CodeMirror { text-align: left; }
.CodeMirror-placeholder { opacity: 0.3; }
.CodeMirror pre { padding: 0px 4px; }
.CodeMirror-lines { padding: 0px; }
div.hr:focus { cursor: none; }
#write pre { white-space: pre-wrap; }
#write.fences-no-line-wrapping pre { white-space: pre; }
#write pre.ty-contain-cm { white-space: normal; }
.CodeMirror-gutters { margin-right: 4px; }
.md-fences { font-size: 0.9rem; display: block; break-inside: avoid; text-align: left; overflow: visible; white-space: pre; background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; position: relative !important; background-position: inherit; background-repeat: inherit; }
.md-fences-adv-panel { width: 100%; margin-top: 10px; text-align: center; padding-top: 0px; padding-bottom: 8px; overflow-x: auto; }
#write .md-fences.mock-cm { white-space: pre-wrap; }
.md-fences.md-fences-with-lineno { padding-left: 0px; }
#write.fences-no-line-wrapping .md-fences.mock-cm { white-space: pre; overflow-x: auto; }
.md-fences.mock-cm.md-fences-with-lineno { padding-left: 8px; }
.CodeMirror-line, twitterwidget { break-inside: avoid; }
.footnotes { opacity: 0.8; font-size: 0.9rem; margin-top: 1em; margin-bottom: 1em; }
.footnotes + .footnotes { margin-top: 0px; }
.md-reset { margin: 0px; padding: 0px; border: 0px; outline: 0px; vertical-align: top; text-decoration: none; text-shadow: none; float: none; position: static; width: auto; height: auto; white-space: nowrap; cursor: inherit; line-height: normal; font-weight: 400; text-align: left; box-sizing: content-box; direction: ltr; background-position: 0px 0px; }
li div { padding-top: 0px; }
blockquote { margin: 1rem 0px; }
li .mathjax-block, li p { margin: 0.5rem 0px; }
li blockquote { margin: 1rem 0px; }
li { margin: 0px; position: relative; }
blockquote > :last-child { margin-bottom: 0px; }
blockquote > :first-child, li > :first-child { margin-top: 0px; }
.footnotes-area { color: rgb(136, 136, 136); margin-top: 0.714rem; padding-bottom: 0.143rem; white-space: normal; }
#write .footnote-line { white-space: pre-wrap; }
@media print { 
  body, html { border: 1px solid transparent; height: 99%; break-after: avoid; break-before: avoid; font-variant-ligatures: no-common-ligatures; }
  #write { margin-top: 0px; padding-top: 0px; border-color: transparent !important; }
  .typora-export * { -webkit-print-color-adjust: exact; }
  .typora-export #write { break-after: avoid; }
  .typora-export #write::after { height: 0px; }
  .is-mac table { break-inside: avoid; }
  .typora-export-show-outline .typora-export-sidebar { display: none; }
}
.footnote-line { margin-top: 0.714em; font-size: 0.7em; }
a img, img a { cursor: pointer; }
pre.md-meta-block { font-size: 0.8rem; min-height: 0.8rem; white-space: pre-wrap; background-color: rgb(204, 204, 204); display: block; overflow-x: hidden; }
p > .md-image:only-child:not(.md-img-error) img, p > img:only-child { display: block; margin: auto; }
#write.first-line-indent p > .md-image:only-child:not(.md-img-error) img { left: -2em; position: relative; }
p > .md-image:only-child { display: inline-block; width: 100%; }
#write .MathJax_Display { margin: 0.8em 0px 0px; }
.md-math-block { width: 100%; }
.md-math-block:not(:empty)::after { display: none; }
.MathJax_ref { fill: currentcolor; }
[contenteditable="true"]:active, [contenteditable="true"]:focus, [contenteditable="false"]:active, [contenteditable="false"]:focus { outline: 0px; box-shadow: none; }
.md-task-list-item { position: relative; list-style-type: none; }
.task-list-item.md-task-list-item { padding-left: 0px; }
.md-task-list-item > input { position: absolute; top: 0px; left: 0px; margin-left: -1.2em; margin-top: calc(1em - 10px); border: none; }
.math { font-size: 1rem; }
.md-toc { min-height: 3.58rem; position: relative; font-size: 0.9rem; border-top-left-radius: 10px; border-top-right-radius: 10px; border-bottom-right-radius: 10px; border-bottom-left-radius: 10px; }
.md-toc-content { position: relative; margin-left: 0px; }
.md-toc-content::after, .md-toc::after { display: none; }
.md-toc-item { display: block; color: rgb(65, 131, 196); }
.md-toc-item a { text-decoration: none; }
.md-toc-inner:hover { text-decoration: underline; }
.md-toc-inner { display: inline-block; cursor: pointer; }
.md-toc-h1 .md-toc-inner { margin-left: 0px; font-weight: 700; }
.md-toc-h2 .md-toc-inner { margin-left: 2em; }
.md-toc-h3 .md-toc-inner { margin-left: 4em; }
.md-toc-h4 .md-toc-inner { margin-left: 6em; }
.md-toc-h5 .md-toc-inner { margin-left: 8em; }
.md-toc-h6 .md-toc-inner { margin-left: 10em; }
@media screen and (max-width: 48em) { 
  .md-toc-h3 .md-toc-inner { margin-left: 3.5em; }
  .md-toc-h4 .md-toc-inner { margin-left: 5em; }
  .md-toc-h5 .md-toc-inner { margin-left: 6.5em; }
  .md-toc-h6 .md-toc-inner { margin-left: 8em; }
}
a.md-toc-inner { font-size: inherit; font-style: inherit; font-weight: inherit; line-height: inherit; }
.footnote-line a:not(.reversefootnote) { color: inherit; }
.md-attr { display: none; }
.md-fn-count::after { content: "."; }
code, pre, samp, tt { font-family: var(--monospace); }
kbd { margin: 0px 0.1em; padding: 0.1em 0.6em; font-size: 0.8em; color: rgb(36, 39, 41); background-color: rgb(255, 255, 255); border: 1px solid rgb(173, 179, 185); border-top-left-radius: 3px; border-top-right-radius: 3px; border-bottom-right-radius: 3px; border-bottom-left-radius: 3px; box-shadow: rgba(12, 13, 14, 0.2) 0px 1px 0px, rgb(255, 255, 255) 0px 0px 0px 2px inset; white-space: nowrap; vertical-align: middle; }
.md-comment { color: rgb(162, 127, 3); opacity: 0.6; font-family: var(--monospace); }
code { text-align: left; }
a.md-print-anchor { white-space: pre !important; border: none !important; display: inline-block !important; position: absolute !important; width: 1px !important; right: 0px !important; outline: 0px !important; text-shadow: initial !important; background-position: 0px 0px !important; }
.os-windows.monocolor-emoji .md-emoji { font-family: "Segoe UI Symbol", sans-serif; }
.md-diagram-panel > svg { max-width: 100%; }
[lang="flow"] svg, [lang="mermaid"] svg { max-width: 100%; height: auto; }
[lang="mermaid"] .node text { font-size: 1rem; }
table tr th { border-bottom-width: 0px; }
video { max-width: 100%; display: block; margin: 0px auto; }
iframe { max-width: 100%; width: 100%; border: none; }
.highlight td, .highlight tr { border: 0px; }
mark { background-color: rgb(255, 255, 0); color: rgb(0, 0, 0); }
.md-html-inline .md-plain, .md-html-inline strong, mark .md-inline-math, mark strong { color: inherit; }
.md-expand mark .md-meta { opacity: 0.3 !important; }
mark .md-meta { color: rgb(0, 0, 0); }
@media print { 
  .typora-export h1, .typora-export h2, .typora-export h3, .typora-export h4, .typora-export h5, .typora-export h6 { break-inside: avoid; }
}
.md-diagram-panel .messageText { stroke: none !important; }
.md-diagram-panel .start-state { fill: var(--node-fill); }
.md-diagram-panel .edgeLabel rect { opacity: 1 !important; }
.md-fences.md-fences-math { font-size: 1em; }
.md-fences-advanced:not(.md-focus) { padding: 0px; white-space: nowrap; border: 0px; }
.md-fences-advanced:not(.md-focus) { background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; background-position: inherit; background-repeat: inherit; }
.typora-export-show-outline .typora-export-content { max-width: 1440px; margin: auto; display: flex; flex-direction: row; }
.typora-export-sidebar { width: 300px; font-size: 0.8rem; margin-top: 80px; margin-right: 18px; }
.typora-export-show-outline #write { --webkit-flex: 2; flex: 2 1 0%; }
.typora-export-sidebar .outline-content { position: fixed; top: 0px; max-height: 100%; overflow: hidden auto; padding-bottom: 30px; padding-top: 60px; width: 300px; }
@media screen and (max-width: 1024px) { 
  .typora-export-sidebar, .typora-export-sidebar .outline-content { width: 240px; }
}
@media screen and (max-width: 800px) { 
  .typora-export-sidebar { display: none; }
}
.outline-content li, .outline-content ul { margin-left: 0px; margin-right: 0px; padding-left: 0px; padding-right: 0px; list-style: none; }
.outline-content ul { margin-top: 0px; margin-bottom: 0px; }
.outline-content strong { font-weight: 400; }
.outline-expander { width: 1rem; height: 1.428571429rem; position: relative; display: table-cell; vertical-align: middle; cursor: pointer; padding-left: 4px; }
.outline-expander::before { content: ""; position: relative; font-family: Ionicons; display: inline-block; font-size: 8px; vertical-align: middle; }
.outline-item { padding-top: 3px; padding-bottom: 3px; cursor: pointer; }
.outline-expander:hover::before { content: ""; }
.outline-h1 > .outline-item { padding-left: 0px; }
.outline-h2 > .outline-item { padding-left: 1em; }
.outline-h3 > .outline-item { padding-left: 2em; }
.outline-h4 > .outline-item { padding-left: 3em; }
.outline-h5 > .outline-item { padding-left: 4em; }
.outline-h6 > .outline-item { padding-left: 5em; }
.outline-label { cursor: pointer; display: table-cell; vertical-align: middle; text-decoration: none; color: inherit; }
.outline-label:hover { text-decoration: underline; }
.outline-item:hover { border-color: rgb(245, 245, 245); background-color: var(--item-hover-bg-color); }
.outline-item:hover { margin-left: -28px; margin-right: -28px; border-left-width: 28px; border-left-style: solid; border-left-color: transparent; border-right-width: 28px; border-right-style: solid; border-right-color: transparent; }
.outline-item-single .outline-expander::before, .outline-item-single .outline-expander:hover::before { display: none; }
.outline-item-open > .outline-item > .outline-expander::before { content: ""; }
.outline-children { display: none; }
.info-panel-tab-wrapper { display: none; }
.outline-item-open > .outline-children { display: block; }
.typora-export .outline-item { padding-top: 1px; padding-bottom: 1px; }
.typora-export .outline-item:hover { margin-right: -8px; border-right-width: 8px; border-right-style: solid; border-right-color: transparent; }
.typora-export .outline-expander::before { content: "+"; font-family: inherit; top: -1px; }
.typora-export .outline-expander:hover::before, .typora-export .outline-item-open > .outline-item > .outline-expander::before { content: "−"; }
.typora-export-collapse-outline .outline-children { display: none; }
.typora-export-collapse-outline .outline-item-open > .outline-children, .typora-export-no-collapse-outline .outline-children { display: block; }
.typora-export-no-collapse-outline .outline-expander::before { content: "" !important; }
.typora-export-show-outline .outline-item-active > .outline-item .outline-label { font-weight: 700; }
.md-inline-math-container mjx-container { zoom: 0.95; }


.CodeMirror { height: auto; }
.CodeMirror.cm-s-inner { background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; background-position: inherit; background-repeat: inherit; }
.CodeMirror-scroll { overflow: auto hidden; z-index: 3; }
.CodeMirror-gutter-filler, .CodeMirror-scrollbar-filler { background-color: rgb(255, 255, 255); }
.CodeMirror-gutters { border-right-width: 1px; border-right-style: solid; border-right-color: rgb(221, 221, 221); background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; white-space: nowrap; background-position: inherit; background-repeat: inherit; }
.CodeMirror-linenumber { padding: 0px 3px 0px 5px; text-align: right; color: rgb(153, 153, 153); }
.cm-s-inner .cm-keyword { color: rgb(119, 0, 136); }
.cm-s-inner .cm-atom, .cm-s-inner.cm-atom { color: rgb(34, 17, 153); }
.cm-s-inner .cm-number { color: rgb(17, 102, 68); }
.cm-s-inner .cm-def { color: rgb(0, 0, 255); }
.cm-s-inner .cm-variable { color: rgb(0, 0, 0); }
.cm-s-inner .cm-variable-2 { color: rgb(0, 85, 170); }
.cm-s-inner .cm-variable-3 { color: rgb(0, 136, 85); }
.cm-s-inner .cm-string { color: rgb(170, 17, 17); }
.cm-s-inner .cm-property { color: rgb(0, 0, 0); }
.cm-s-inner .cm-operator { color: rgb(152, 26, 26); }
.cm-s-inner .cm-comment, .cm-s-inner.cm-comment { color: rgb(170, 85, 0); }
.cm-s-inner .cm-string-2 { color: rgb(255, 85, 0); }
.cm-s-inner .cm-meta { color: rgb(85, 85, 85); }
.cm-s-inner .cm-qualifier { color: rgb(85, 85, 85); }
.cm-s-inner .cm-builtin { color: rgb(51, 0, 170); }
.cm-s-inner .cm-bracket { color: rgb(153, 153, 119); }
.cm-s-inner .cm-tag { color: rgb(17, 119, 0); }
.cm-s-inner .cm-attribute { color: rgb(0, 0, 204); }
.cm-s-inner .cm-header, .cm-s-inner.cm-header { color: rgb(0, 0, 255); }
.cm-s-inner .cm-quote, .cm-s-inner.cm-quote { color: rgb(0, 153, 0); }
.cm-s-inner .cm-hr, .cm-s-inner.cm-hr { color: rgb(153, 153, 153); }
.cm-s-inner .cm-link, .cm-s-inner.cm-link { color: rgb(0, 0, 204); }
.cm-negative { color: rgb(221, 68, 68); }
.cm-positive { color: rgb(34, 153, 34); }
.cm-header, .cm-strong { font-weight: 700; }
.cm-del { text-decoration: line-through; }
.cm-em { font-style: italic; }
.cm-link { text-decoration: underline; }
.cm-error { color: red; }
.cm-invalidchar { color: red; }
.cm-constant { color: rgb(38, 139, 210); }
.cm-defined { color: rgb(181, 137, 0); }
div.CodeMirror span.CodeMirror-matchingbracket { color: rgb(0, 255, 0); }
div.CodeMirror span.CodeMirror-nonmatchingbracket { color: rgb(255, 34, 34); }
.cm-s-inner .CodeMirror-activeline-background { background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; background-position: inherit; background-repeat: inherit; }
.CodeMirror { position: relative; overflow: hidden; }
.CodeMirror-scroll { height: 100%; outline: 0px; position: relative; box-sizing: content-box; background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; background-position: inherit; background-repeat: inherit; }
.CodeMirror-sizer { position: relative; }
.CodeMirror-gutter-filler, .CodeMirror-hscrollbar, .CodeMirror-scrollbar-filler, .CodeMirror-vscrollbar { position: absolute; z-index: 6; display: none; outline: 0px; }
.CodeMirror-vscrollbar { right: 0px; top: 0px; overflow: hidden; }
.CodeMirror-hscrollbar { bottom: 0px; left: 0px; overflow: auto hidden; }
.CodeMirror-scrollbar-filler { right: 0px; bottom: 0px; }
.CodeMirror-gutter-filler { left: 0px; bottom: 0px; }
.CodeMirror-gutters { position: absolute; left: 0px; top: 0px; padding-bottom: 10px; z-index: 3; overflow-y: hidden; }
.CodeMirror-gutter { white-space: normal; height: 100%; box-sizing: content-box; padding-bottom: 30px; margin-bottom: -32px; display: inline-block; }
.CodeMirror-gutter-wrapper { position: absolute; z-index: 4; border: none !important; background-position: 0px 0px !important; }
.CodeMirror-gutter-background { position: absolute; top: 0px; bottom: 0px; z-index: 4; }
.CodeMirror-gutter-elt { position: absolute; cursor: default; z-index: 4; }
.CodeMirror-lines { cursor: text; }
.CodeMirror pre { border-top-left-radius: 0px; border-top-right-radius: 0px; border-bottom-right-radius: 0px; border-bottom-left-radius: 0px; border-width: 0px; font-family: inherit; font-size: inherit; margin: 0px; white-space: pre; word-wrap: normal; color: inherit; z-index: 2; position: relative; overflow: visible; background-position: 0px 0px; }
.CodeMirror-wrap pre { word-wrap: break-word; white-space: pre-wrap; word-break: normal; }
.CodeMirror-code pre { border-right-width: 30px; border-right-style: solid; border-right-color: transparent; width: fit-content; }
.CodeMirror-wrap .CodeMirror-code pre { border-right-style: none; width: auto; }
.CodeMirror-linebackground { position: absolute; inset: 0px; z-index: 0; }
.CodeMirror-linewidget { position: relative; z-index: 2; overflow: auto; }
.CodeMirror-wrap .CodeMirror-scroll { overflow-x: hidden; }
.CodeMirror-measure { position: absolute; width: 100%; height: 0px; overflow: hidden; visibility: hidden; }
.CodeMirror-measure pre { position: static; }
.CodeMirror div.CodeMirror-cursor { position: absolute; visibility: hidden; border-right-style: none; width: 0px; }
.CodeMirror div.CodeMirror-cursor { visibility: hidden; }
.CodeMirror-focused div.CodeMirror-cursor { visibility: inherit; }
.cm-searching { background-color: rgba(255, 255, 0, 0.4); }
span.cm-underlined { text-decoration: underline; }
span.cm-strikethrough { text-decoration: line-through; }
.cm-tw-syntaxerror { color: rgb(255, 255, 255); background-color: rgb(153, 0, 0); }
.cm-tw-deleted { text-decoration: line-through; }
.cm-tw-header5 { font-weight: 700; }
.cm-tw-listitem:first-child { padding-left: 10px; }
.cm-tw-box { border-style: solid; border-right-width: 1px; border-bottom-width: 1px; border-left-width: 1px; border-color: inherit; border-top-width: 0px !important; }
.cm-tw-underline { text-decoration: underline; }
@media print { 
  .CodeMirror div.CodeMirror-cursor { visibility: hidden; }
}


:root {
    --side-bar-bg-color: #fafafa;
    --control-text-color: #777;
}

@include-when-export url(https://fonts.loli.net/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext);

/* open-sans-regular - latin-ext_latin */
  /* open-sans-italic - latin-ext_latin */
    /* open-sans-700 - latin-ext_latin */
    /* open-sans-700italic - latin-ext_latin */
  html {
    font-size: 16px;
    -webkit-font-smoothing: antialiased;
}

body {
    font-family: "Open Sans","Clear Sans", "Helvetica Neue", Helvetica, Arial, 'Segoe UI Emoji', sans-serif;
    color: rgb(51, 51, 51);
    line-height: 1.6;
}

#write {
    max-width: 860px;
  	margin: 0 auto;
  	padding: 30px;
    padding-bottom: 100px;
}

@media only screen and (min-width: 1400px) {
	#write {
		max-width: 1024px;
	}
}

@media only screen and (min-width: 1800px) {
	#write {
		max-width: 1200px;
	}
}

#write > ul:first-child,
#write > ol:first-child{
    margin-top: 30px;
}

a {
    color: #4183C4;
}
h1,
h2,
h3,
h4,
h5,
h6 {
    position: relative;
    margin-top: 1rem;
    margin-bottom: 1rem;
    font-weight: bold;
    line-height: 1.4;
    cursor: text;
}
h1:hover a.anchor,
h2:hover a.anchor,
h3:hover a.anchor,
h4:hover a.anchor,
h5:hover a.anchor,
h6:hover a.anchor {
    text-decoration: none;
}
h1 tt,
h1 code {
    font-size: inherit;
}
h2 tt,
h2 code {
    font-size: inherit;
}
h3 tt,
h3 code {
    font-size: inherit;
}
h4 tt,
h4 code {
    font-size: inherit;
}
h5 tt,
h5 code {
    font-size: inherit;
}
h6 tt,
h6 code {
    font-size: inherit;
}
h1 {
    font-size: 2.25em;
    line-height: 1.2;
    border-bottom: 1px solid #eee;
}
h2 {
    font-size: 1.75em;
    line-height: 1.225;
    border-bottom: 1px solid #eee;
}

/*@media print {
    .typora-export h1,
    .typora-export h2 {
        border-bottom: none;
        padding-bottom: initial;
    }

    .typora-export h1::after,
    .typora-export h2::after {
        content: "";
        display: block;
        height: 100px;
        margin-top: -96px;
        border-top: 1px solid #eee;
    }
}*/

h3 {
    font-size: 1.5em;
    line-height: 1.43;
}
h4 {
    font-size: 1.25em;
}
h5 {
    font-size: 1em;
}
h6 {
   font-size: 1em;
    color: #777;
}
p,
blockquote,
ul,
ol,
dl,
table{
    margin: 0.8em 0;
}
li>ol,
li>ul {
    margin: 0 0;
}
hr {
    height: 2px;
    padding: 0;
    margin: 16px 0;
    background-color: #e7e7e7;
    border: 0 none;
    overflow: hidden;
    box-sizing: content-box;
}

li p.first {
    display: inline-block;
}
ul,
ol {
    padding-left: 30px;
}
ul:first-child,
ol:first-child {
    margin-top: 0;
}
ul:last-child,
ol:last-child {
    margin-bottom: 0;
}
blockquote {
    border-left: 4px solid #dfe2e5;
    padding: 0 15px;
    color: #777777;
}
blockquote blockquote {
    padding-right: 0;
}
table {
    padding: 0;
    word-break: initial;
}
table tr {
    border: 1px solid #dfe2e5;
    margin: 0;
    padding: 0;
}
table tr:nth-child(2n),
thead {
    background-color: #f8f8f8;
}
table th {
    font-weight: bold;
    border: 1px solid #dfe2e5;
    border-bottom: 0;
    margin: 0;
    padding: 6px 13px;
}
table td {
    border: 1px solid #dfe2e5;
    margin: 0;
    padding: 6px 13px;
}
table th:first-child,
table td:first-child {
    margin-top: 0;
}
table th:last-child,
table td:last-child {
    margin-bottom: 0;
}

.CodeMirror-lines {
    padding-left: 4px;
}

.code-tooltip {
    box-shadow: 0 1px 1px 0 rgba(0,28,36,.3);
    border-top: 1px solid #eef2f2;
}

.md-fences,
code,
tt {
    border: 1px solid #e7eaed;
    background-color: #f8f8f8;
    border-radius: 3px;
    padding: 0;
    padding: 2px 4px 0px 4px;
    font-size: 0.9em;
}

code {
    background-color: #f3f4f4;
    padding: 0 2px 0 2px;
}

.md-fences {
    margin-bottom: 15px;
    margin-top: 15px;
    padding-top: 8px;
    padding-bottom: 6px;
}


.md-task-list-item > input {
  margin-left: -1.3em;
}

@media print {
    html {
        font-size: 13px;
    }
    table,
    pre {
        page-break-inside: avoid;
    }
    pre {
        word-wrap: break-word;
    }
}

.md-fences {
	background-color: #f8f8f8;
}
#write pre.md-meta-block {
	padding: 1rem;
    font-size: 85%;
    line-height: 1.45;
    background-color: #f7f7f7;
    border: 0;
    border-radius: 3px;
    color: #777777;
    margin-top: 0 !important;
}

.mathjax-block>.code-tooltip {
	bottom: .375rem;
}

.md-mathjax-midline {
    background: #fafafa;
}

#write>h3.md-focus:before{
	left: -1.5625rem;
	top: .375rem;
}
#write>h4.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
#write>h5.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
#write>h6.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
.md-image>.md-meta {
    /*border: 1px solid #ddd;*/
    border-radius: 3px;
    padding: 2px 0px 0px 4px;
    font-size: 0.9em;
    color: inherit;
}

.md-tag {
    color: #a7a7a7;
    opacity: 1;
}

.md-toc { 
    margin-top:20px;
    padding-bottom:20px;
}

.sidebar-tabs {
    border-bottom: none;
}

#typora-quick-open {
    border: 1px solid #ddd;
    background-color: #f8f8f8;
}

#typora-quick-open-item {
    background-color: #FAFAFA;
    border-color: #FEFEFE #e5e5e5 #e5e5e5 #eee;
    border-style: solid;
    border-width: 1px;
}

/** focus mode */
.on-focus-mode blockquote {
    border-left-color: rgba(85, 85, 85, 0.12);
}

header, .context-menu, .megamenu-content, footer{
    font-family: "Segoe UI", "Arial", sans-serif;
}

.file-node-content:hover .file-node-icon,
.file-node-content:hover .file-node-open-state{
    visibility: visible;
}

.mac-seamless-mode #typora-sidebar {
    background-color: #fafafa;
    background-color: var(--side-bar-bg-color);
}

.md-lang {
    color: #b4654d;
}

/*.html-for-mac {
    --item-hover-bg-color: #E6F0FE;
}*/

#md-notification .btn {
    border: 0;
}

.dropdown-menu .divider {
    border-color: #e5e5e5;
    opacity: 0.4;
}

.ty-preferences .window-content {
    background-color: #fafafa;
}

.ty-preferences .nav-group-item.active {
    color: white;
    background: #999;
}

.menu-item-container a.menu-style-btn {
    background-color: #f5f8fa;
    background-image: linear-gradient( 180deg , hsla(0, 0%, 100%, 0.8), hsla(0, 0%, 100%, 0)); 
}


 @media print { @page {margin: 0 0 0 0;} body.typora-export {padding-left: 0; padding-right: 0;} #write {padding:0;}}
</style><title>10_hydraRegulators</title>
</head>
<body class='typora-export'><div class='typora-export-content'>
<div id='write'  class=''><h1 id='identifying-candidate-regulators-of-cell-type-specific-transcription-in-hydra'><span>Identifying Candidate Regulators of Cell-Type-Specific Transcription in </span><em><span>Hydra</span></em></h1><p><span>This document covers our analysis aimed at identifying transcriptional regulators of gene co-expression modules in the </span><em><span>Hydra</span></em><span> single-cell atlas. This entailed integrating our non-negative matrix factorization (NMF) results (described in </span><code>05_hydraAtlasReMap.md</code><span>) with our transcription factor binding site conservation analysis (described in </span><code>07_genomeConservation.md</code><span>) to identify motifs enriched in the putative regulatory sequences of sets of co-expressed genes (i.e., metagenes). Following the motif enrichment analysis, we then looked for transcription factors whose expression pattern and sequence binding preferences correlated with motif enrichement patterns. </span></p><div class='md-toc' mdtype='toc'><p class="md-toc-content" role="list"><span role="listitem" class="md-toc-item md-toc-h1" data-ref="n0"><a class="md-toc-inner" style="" href="#identifying-candidate-regulators-of-cell-type-specific-transcription-in-hydra">Identifying Candidate Regulators of Cell-Type-Specific Transcription in <em>Hydra</em></a></span><span role="listitem" class="md-toc-item md-toc-h2" data-ref="n4"><a class="md-toc-inner" style="" href="#preparatory-motif-and-transcription-factor-analyses">Preparatory Motif and Transcription Factor Analyses</a></span><span role="listitem" class="md-toc-item md-toc-h3" data-ref="n6"><a class="md-toc-inner" style="" href="#linking-predicted-transcription-factor-sites-to-their-nearest-genes">Linking Predicted Transcription Factor Sites to Their Nearest Genes </a></span><span role="listitem" class="md-toc-item md-toc-h3" data-ref="n16"><a class="md-toc-inner" style="" href="#clustering-motifs-to-reduce-redundancy">Clustering Motifs to Reduce Redundancy</a></span><span role="listitem" class="md-toc-item md-toc-h3" data-ref="n29"><a class="md-toc-inner" style="" href="#annotate-transcription-factors-in-the-genome-gene-models">Annotate Transcription Factors in the Genome Gene Models</a></span><span role="listitem" class="md-toc-item md-toc-h3" data-ref="n34"><a class="md-toc-inner" style="" href="#download-motif-metadata">Download Motif Metadata</a></span><span role="listitem" class="md-toc-item md-toc-h2" data-ref="n41"><a class="md-toc-inner" style="" href="#single-cell-motif-enrichment-analysis">Single-Cell Motif Enrichment Analysis</a></span><span role="listitem" class="md-toc-item md-toc-h3" data-ref="n42"><a class="md-toc-inner" style="" href="#analysis-using-bona-fide-binding-motifs">Analysis Using Bona Fide Binding Motifs</a></span><span role="listitem" class="md-toc-item md-toc-h4" data-ref="n127"><a class="md-toc-inner" style="" href="#performing-gene-set-enrichment-analyses-1">Performing Gene Set Enrichment Analyses</a></span><span role="listitem" class="md-toc-item md-toc-h4" data-ref="n51"><a class="md-toc-inner" style="" href="#generating-single-cell-motif-enrichment-scores-1">Generating Single-Cell Motif Enrichment Scores</a></span><span role="listitem" class="md-toc-item md-toc-h4" data-ref="n58"><a class="md-toc-inner" style="" href="#removing-redundant-motifs-from-the-enrichment-results">Removing Redundant Motifs from the Enrichment Results</a></span><span role="listitem" class="md-toc-item md-toc-h4" data-ref="n69"><a class="md-toc-inner" style="" href="#identifying-candidate-regulators-of-enriched-motifs">Identifying Candidate Regulators of Enriched Motifs</a></span><span role="listitem" class="md-toc-item md-toc-h3" data-ref="n81"><a class="md-toc-inner" style="" href="#analysis-using-shuffled-binding-motifs">Analysis Using Shuffled Binding Motifs</a></span><span role="listitem" class="md-toc-item md-toc-h4" data-ref="n139"><a class="md-toc-inner" style="" href="#performing-gene-set-enrichment-analyses-2">Performing Gene Set Enrichment Analyses</a></span><span role="listitem" class="md-toc-item md-toc-h4" data-ref="n87"><a class="md-toc-inner" style="" href="#generating-single-cell-motif-enrichment-scores-2">Generating Single-Cell Motif Enrichment Scores</a></span><span role="listitem" class="md-toc-item md-toc-h3" data-ref="n95"><a class="md-toc-inner" style="" href="#comparing-enrichment-results-for-genuine-and-shuffled-motifs">Comparing Enrichment Results for Genuine and Shuffled Motifs</a></span><span role="listitem" class="md-toc-item md-toc-h2" data-ref="n120"><a class="md-toc-inner" style="" href="#files-associated-with-this-document">Files Associated with This Document</a></span></p></div><h2 id='preparatory-motif-and-transcription-factor-analyses'><span>Preparatory Motif and Transcription Factor Analyses</span></h2><p><span>Prior to the main analysis, several supplemental files needed to be generated. These included annotations linking conserved motifs to their most proximal gene, information on motif sequence similarity (to reduce redundancy in the enrichment results), the names of the transcription factor associated with each binding motif, and predictions of which gene models in the AEP assembly are transcription factors based on their pfam domain annotations. </span></p><h3 id='linking-predicted-transcription-factor-sites-to-their-nearest-genes'><span>Linking Predicted Transcription Factor Sites to Their Nearest Genes </span></h3><p><span>First, we used UROPA to link conserved transcription factor binding sites to their putative target genes based on proximity. Specifically, we linked a motif to the nearest transcription start site as long as it fell within 30 Kb. We did this for both the binding site predictions based on the bona fide JASPAR motif sequences as well as the predictions based on shuffled motif sequences (both generated in </span><code>07_genomeConservation.md</code><span>).</span></p><p><span>We used the following config file to annotate the binding site predictions for the bona fide motif sequences (</span><code>conMotsATAC.bed</code><span>):</span></p><p><span>(</span><em><span>01_motifPrep/conMotATACAnnot.json</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="json"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="json"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">{</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"queries"</span>:[</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  {<span class="cm-string cm-property">"distance"</span>: [<span class="cm-number">30000</span>, <span class="cm-number">30000</span>], <span class="cm-string cm-property">"feature"</span>: [<span class="cm-string">"gene"</span>], <span class="cm-string cm-property">"feature_anchor"</span>: [<span class="cm-string">"start"</span>], <span class="cm-string cm-property">"name"</span>: <span class="cm-string">"query_1"</span>, <span class="cm-string cm-property">"relative_location"</span>: [<span class="cm-string">"Downstream"</span>, <span class="cm-string">"Upstream"</span>, <span class="cm-string">"OverlapStart"</span>, <span class="cm-string">"FeatureInsidePeak"</span>, <span class="cm-string">"PeakInsideFeature"</span>, <span class="cm-string">"OverlapEnd"</span>], <span class="cm-string cm-property">"strand"</span>: <span class="cm-string">"ignore"</span>}</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp;  ],</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"show_attributes"</span>: [<span class="cm-string">"gene_id"</span>],</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"priority"</span>: <span class="cm-string">"False"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"gtf"</span>: <span class="cm-string">"HVAEP1.GeneModels.longestIso.gtf"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"bed"</span>: <span class="cm-string">"conMotsATAC.bed"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"prefix"</span>: <span class="cm-string">"conMotATAC"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"outdir"</span>: <span class="cm-string">"motifAnnot"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"threads"</span>: <span class="cm-string">"1"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"output_by_query"</span>: <span class="cm-string">"False"</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">}</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 330px;"></div><div class="CodeMirror-gutters" style="display: none; height: 330px;"></div></div></div></pre><p><span>We used the following config file to annotate the binding site predictions for the shuffled motif predictions (</span><code>conShufMotsATAC.bed</code><span>)</span></p><p><span>(</span><em><span>01_motifPrep/conShufMotATACAnnot.json</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="json"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="json"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">{</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"queries"</span>:[</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  {<span class="cm-string cm-property">"distance"</span>: [<span class="cm-number">30000</span>, <span class="cm-number">30000</span>], <span class="cm-string cm-property">"feature"</span>: [<span class="cm-string">"gene"</span>], <span class="cm-string cm-property">"feature_anchor"</span>: [<span class="cm-string">"start"</span>], <span class="cm-string cm-property">"name"</span>: <span class="cm-string">"query_1"</span>, <span class="cm-string cm-property">"relative_location"</span>: [<span class="cm-string">"Downstream"</span>, <span class="cm-string">"Upstream"</span>, <span class="cm-string">"OverlapStart"</span>, <span class="cm-string">"FeatureInsidePeak"</span>, <span class="cm-string">"PeakInsideFeature"</span>, <span class="cm-string">"OverlapEnd"</span>], <span class="cm-string cm-property">"strand"</span>: <span class="cm-string">"ignore"</span>}</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp;  ],</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"show_attributes"</span>: [<span class="cm-string">"gene_id"</span>],</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"priority"</span>: <span class="cm-string">"False"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"gtf"</span>: <span class="cm-string">"HVAEP1.GeneModels.longestIso.gtf"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"bed"</span>: <span class="cm-string">"conShufMotsATAC.bed"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"prefix"</span>: <span class="cm-string">"conShufMotsATAC"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"outdir"</span>: <span class="cm-string">"motifAnnot"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"threads"</span>: <span class="cm-string">"1"</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-string cm-property">"output_by_query"</span>: <span class="cm-string">"False"</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">}</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 330px;"></div><div class="CodeMirror-gutters" style="display: none; height: 330px;"></div></div></div></pre><p><span>We then ran the two annotation analyses with the following commands:</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">uropa <span class="cm-attribute">-i</span> conMotATACAnnot.json </span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">uropa <span class="cm-attribute">-i</span> conShufMotATACAnnot.json</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 44px;"></div><div class="CodeMirror-gutters" style="display: none; height: 44px;"></div></div></div></pre><h3 id='clustering-motifs-to-reduce-redundancy'><span>Clustering Motifs to Reduce Redundancy</span></h3><p><span>Closely related transcription factors that use the same DNA binding domain tend to have highly similar (or virtually identical) binding preferences. Thus, motif enrichment analyses tend to have many redundant motifs in their results. To identify redundant motifs in our compiled JASPAR database, we used two clustering analyses: one that grouped motifs purely based on their sequence composition and another that grouped motifs based on their enrichment patterns in the single-cell atlas. In this section, we describe the former clustering approach (the latter approach is described later). </span></p><p><span>To perform the sequence-based clustering, we chose to make use of the </span><code>compareMotifs.pl</code><span> utility script provided as part of the HOMER suite of motif enrichment tools. In order to use this script, we needed to reformat our JASPAR-formated motifs into the HOMER motif format. We did this using a utility script also provided as part of HOMER:</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">perl parseJasparMatrix.pl pooledJasparNR.txt &gt; pooledJasparNR.homer.motifs</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 22px;"></div><div class="CodeMirror-gutters" style="display: none; height: 22px;"></div></div></div></pre><p><span>We then used the </span><code>compareMotifs.pl</code><span> to generate a matrix of pairwise similarity scores for all motifs in our custom JASPAR database</span></p><p><span>(</span><em><span>01_motifPrep/homerMotCompare.sh</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><span><span>​</span>x</span></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-meta">#!/bin/bash</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">perl compareMotifs.pl pooledJasparNR.homer.motifs motCompare <span class="cm-attribute">-matrix</span> motComp.txt</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 66px;"></div><div class="CodeMirror-gutters" style="display: none; height: 66px;"></div></div></div></pre><p><span>Within a custom R script, we used these similarity scores for a hierarchical clustering analysis that grouped together motifs with similar sequence composition. These results were later combined with a second hierarchical clustering analysis (described in the &quot;Single-Cell Motif Enrichment Analysis&quot; section) to arrive at the final motif cluster assignments.</span></p><p><span>(</span><em><span>01_motifPrep/clusterMots.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">rstudioapi</span>)</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">setwd</span>(<span class="cm-variable">dirname</span>(<span class="cm-variable">getActiveDocumentContext</span>()<span class="cm-operator cm-dollar">$</span><span class="cm-variable">path</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#import similarity matrix for JASPAR animal PWMs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">similarityMat</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.delim</span>(<span class="cm-string">"motComp.txt"</span>, <span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>, <span class="cm-variable">header</span> <span class="cm-operator">=</span> <span class="cm-variable">T</span>, <span class="cm-variable">sep</span> <span class="cm-operator">=</span> <span class="cm-string">"</span><span class="cm-string-2">\t</span><span class="cm-string">"</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#reformat motif names (rows and columns)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">newNames</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">rownames</span>(<span class="cm-variable">similarityMat</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">newNames</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gsub</span>(<span class="cm-string">'/Jaspar'</span>,<span class="cm-string">''</span>,<span class="cm-variable">newNames</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">newNames</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gsub</span>(<span class="cm-string">'/'</span>,<span class="cm-string">'_'</span>,<span class="cm-variable">newNames</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rownames</span>(<span class="cm-variable">similarityMat</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">newNames</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">colnames</span>(<span class="cm-variable">similarityMat</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">newNames</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#generate distance matrix from correlation coefficient</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">d</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">as.dist</span>(<span class="cm-number">1</span> <span class="cm-operator">-</span> <span class="cm-variable">similarityMat</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#use hierarchical clustering to group motifs based on correlation</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">hc1</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">hclust</span>(<span class="cm-variable">d</span>, <span class="cm-variable">method</span> <span class="cm-operator">=</span> <span class="cm-string">"average"</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pdf</span>(<span class="cm-string">'motDistClust.pdf'</span>,<span class="cm-variable">width</span><span class="cm-operator">=</span><span class="cm-number">150</span>,<span class="cm-variable">height</span><span class="cm-operator">=</span><span class="cm-number">20</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">plot</span>(<span class="cm-variable">hc1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rect.hclust</span>(<span class="cm-variable">hc1</span> , <span class="cm-variable">h</span> <span class="cm-operator">=</span> <span class="cm-number">0.3</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">abline</span>(<span class="cm-variable">h</span> <span class="cm-operator">=</span> <span class="cm-number">0.3</span>, <span class="cm-variable">col</span> <span class="cm-operator">=</span> <span class="cm-string">'red'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">dev.off</span>()</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#cut the branches at 0.3 and group accordingly</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">clusts</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">cutree</span>(<span class="cm-variable">hc1</span>, <span class="cm-variable">h</span> <span class="cm-operator">=</span> <span class="cm-number">0.3</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">length</span>(<span class="cm-variable">unique</span>(<span class="cm-variable">clusts</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#save cluster information</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">repMotifs</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">data.frame</span>(<span class="cm-variable">ID</span> <span class="cm-operator">=</span> <span class="cm-variable">names</span>(<span class="cm-variable">clusts</span>), <span class="cm-variable">clust</span> <span class="cm-operator">=</span> <span class="cm-variable">clusts</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">write.csv</span>(<span class="cm-variable">repMotifs</span>, <span class="cm-variable">file</span> <span class="cm-operator">=</span> <span class="cm-string">"motif_clusters.csv"</span>)</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 770px;"></div><div class="CodeMirror-gutters" style="display: none; height: 770px;"></div></div></div></pre><p><span>Below is the tree generated by the hierarchical clustering analysis used to generate the </span><code>motif_clusters.csv</code><span> file:</span></p><p><img src="resources/motDistClust.png" referrerpolicy="no-referrer" alt="motClust"></p><p>&nbsp;</p><h3 id='annotate-transcription-factors-in-the-genome-gene-models'><span>Annotate Transcription Factors in the Genome Gene Models</span></h3><p><span>Ultimately, the goal of the analysis was to assign transcription factors as candidate regulators of co-expressed genes. This required that we determine which gene models in the AEP assembly are likely to be transcription factors. To do this, we made use of our InterProScan results (described in </span><code>03_aepGenomeAnnotation.md</code><span>). Specifically, we used a manually curated list of protein domains and gene ontology terms to generate a candidate list of transcription factors.</span></p><p><span>(</span><em><span>01_motifPrep/tfList.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">rstudioapi</span>)</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">setwd</span>(<span class="cm-variable">dirname</span>(<span class="cm-variable">getActiveDocumentContext</span>()<span class="cm-operator cm-dollar">$</span><span class="cm-variable">path</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#import ipr results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ipr</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.delim</span>(<span class="cm-string">'HVAEP1.prot.longestIso.fa.tsv'</span>, <span class="cm-variable">header</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#pull initial list of candidate TFs based on either domain composition or GO annotation</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ipr.tf</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">ipr</span>[<span class="cm-variable">grepl</span>(<span class="cm-string">'GO:0003700|GO:0003677|GO:0006355|GO:0043565'</span>,<span class="cm-variable">ipr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V14</span>) <span class="cm-operator">|</span> <span class="cm-variable">grepl</span>(<span class="cm-string">'helix-loop-helix|hmg|sox|winged|c2h2'</span>,<span class="cm-variable">ipr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V6</span>,<span class="cm-variable">ignore.case</span> <span class="cm-operator">=</span> <span class="cm-variable">T</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop genes associated with non-TF GO terms</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">exclIDs</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">ipr</span>[<span class="cm-variable">grepl</span>(<span class="cm-string">'GO:0000814|GO:0008180|GO:0005840|GO:0031213|GO:0032021|GO:0006281|GO:0000150|GO:0003721|GO:0030337|GO:0006298|GO:0000723|GO:0006302|GO:0003899|GO:0006351|GO:0006313|GO:0003887|GO:0000786|GO:0003697|GO:0006260|GO:0016592|GO:0003755|GO:0006265|GO:0003684|GO:0006338|GO:0006384|GO:0006383|GO:0003723|GO:0006270|GO:0003910|GO:0006367|GO:0032508|GO:0006289'</span>,<span class="cm-variable">ipr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V14</span>),<span class="cm-number">1</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop genes associated with non-TF annotations</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">moreEclIDs</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">ipr</span>[<span class="cm-variable">grepl</span>(<span class="cm-string">'Oxygenase|transferase|Transposase|phosphodiesterase|peptidase|DNA REPLICATION|Tetratricopeptide|membrane-bound|hydrolase|proteinase|mitochondrial|reductase|synthase|autointegration|nuclease|TRANSFERASE|histone|helicase|topoisomerase|DNA polymerase|rejoining|rad51|Membrane-anchored|Translin|BESS|seet_6|LIPOMA|QSOX'</span>,<span class="cm-variable">ipr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V6</span>, <span class="cm-variable">ignore.case</span> <span class="cm-operator">=</span> <span class="cm-variable">T</span>),<span class="cm-number">1</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ipr.tf</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">ipr.tf</span>[<span class="cm-operator">!</span>(<span class="cm-variable">ipr.tf</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V1</span> <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">c</span>(<span class="cm-variable">exclIDs</span>,<span class="cm-variable">moreEclIDs</span>)),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#subset to just the IDs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">tfIDs</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">unique</span>(<span class="cm-variable">ipr.tf</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#convert transcript IDs to gene IDs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">tfIDs</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gsub</span>(<span class="cm-string">'HVAEP1_T(</span><span class="cm-string-2">\\</span><span class="cm-string">d+)[.]</span><span class="cm-string-2">\\</span><span class="cm-string">d+'</span>,<span class="cm-string">'HVAEP1_G</span><span class="cm-string-2">\\</span><span class="cm-string">1'</span>,<span class="cm-variable">tfIDs</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">write.csv</span>(<span class="cm-variable">ipr.tf</span>,<span class="cm-string">'fullTF.csv'</span>,<span class="cm-variable">row.names</span><span class="cm-operator">=</span><span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">write.table</span>(<span class="cm-variable">tfIDs</span>,<span class="cm-string">'tfIDs.txt'</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>,<span class="cm-variable">col.names</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>,<span class="cm-builtin">quote</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 770px;"></div><div class="CodeMirror-gutters" style="display: none; height: 770px;"></div></div></div></pre><p><span>Based on this analysis, we identified a total of 811 candidate transcription factors in the AEP gene models. </span></p><h3 id='download-motif-metadata'><span>Download Motif Metadata</span></h3><p><span>In order to determine which transcription factors could bind which JASPAR motifs, we made use of the uniprot database protein domain annotations available for each JASPAR motif. JASPAR provides uniprot IDs for each motif in its database, and each uniprot entry contains Pfam domain annotations. This allowed us to determine the pfam domains associated with each motif in our motif database. These domains could then be linked to the domain predictions from our InterProScan analysis to associate specific gene models with JASPAR motifs. These gene/motif links provided the basis for predicting regulators of gene co-expression.</span></p><p><span>To access the metadata for each JASPAR motif, we first had to compile a list of all JASPAR motif IDs of interest:</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">grep</span> <span class="cm-string">'&gt;'</span> pooledJasparNR.txt | <span class="cm-builtin">sed</span> <span class="cm-string">'s/&gt;\([^\t]\+\).*/\1/g'</span> &gt; jasparIDs.txt</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 22px;"></div><div class="CodeMirror-gutters" style="display: none; height: 22px;"></div></div></div></pre><p><span>We then used both the JASPAR and uniprot REST APIs to download the Pfam domains associated with each JASPAR motif. We then cross-referenced these domains with the domain composition of our list of predicted transcription factor gene models to generate lists of genes that could plausibly bind to each binding motif. </span></p><p><span>(</span><em><span>01_motifPrep/protDBLink.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="bash" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="bash"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">library(rstudioapi)</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">library(jsonlite)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">library(httr)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">library(RCurl)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">setwd(dirname(getActiveDocumentContext()<span class="cm-def">$path</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#get full list of jaspar motif IDs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">motif.IDs &lt;- read.delim(<span class="cm-string">'jasparIDs.txt'</span>)[,1]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">results &lt;- list(length(motif.IDs))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#get TF name and family for each motif ID</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-keyword">for</span> (i <span class="cm-keyword">in</span> <span class="cm-number">1</span>:length(motif.IDs)) {</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  print(i)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  link &lt;- paste0(<span class="cm-string">"http://jaspar2020.genereg.net/api/v1/matrix/"</span>,as.character(motif.IDs[i]),<span class="cm-string">"/"</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  result &lt;- fromJSON(url(link))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  result.n &lt;- paste(result[[c(<span class="cm-string">"name"</span>)]],collapse <span class="cm-operator">=</span> <span class="cm-string">', '</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  result.f &lt;- paste(result[[c(<span class="cm-string">"family"</span>)]],collapse <span class="cm-operator">=</span> <span class="cm-string">', '</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  results[[i]] &lt;- c(result.n,result.f)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">}</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#collapse results into DF</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">motInfo &lt;- data.frame<span class="cm-def">(ID</span><span class="cm-operator">=</span>motif.IDs,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-def">name</span><span class="cm-operator">=</span>vapply(results,function(x) x[1],<span class="cm-string">""</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-def">family</span><span class="cm-operator">=</span>vapply(results,function(x) x[2],<span class="cm-string">""</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">write</span>.csv(motInfo,<span class="cm-string">'motifInfo.csv'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#get uniprot IDs for each motif from JASPAR</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">results &lt;- list(length(motif.IDs))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-keyword">for</span> (i <span class="cm-keyword">in</span> <span class="cm-number">1</span>:length(motif.IDs)) {</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  print(i)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  link &lt;- paste0(<span class="cm-string">"http://jaspar2020.genereg.net/api/v1/matrix/"</span>,as.character(motif.IDs[i]),<span class="cm-string">"/"</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  result &lt;- fromJSON(url(link))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  result &lt;- result[[<span class="cm-string">"uniprot_ids"</span>]]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-keyword">if</span>(length(result !<span class="cm-operator">=</span> <span class="cm-number">0</span>)) {</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  results[[i]] &lt;- result</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  } <span class="cm-keyword">else</span> {</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  print(<span class="cm-string">"Empty"</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  }</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">}</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#pull names for each result</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">names(results) &lt;- motif.IDs</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">pfamAnnot &lt;- list()</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#Pull the pfam domains assocaited with each swissprot entry</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-keyword">for</span> (i <span class="cm-keyword">in</span> <span class="cm-number">1</span>:length(results)) {</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  print(i)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  subRes &lt;- results[[i]]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  enGene &lt;- c()</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-keyword">for</span>(ID <span class="cm-keyword">in</span> subRes){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  print(ID)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  link &lt;- paste0(<span class="cm-string">"https://www.uniprot.org/uniprot/"</span>,ID,<span class="cm-string">".txt"</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  result &lt;- GET(link)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  result &lt;- rawToChar(result<span class="cm-def">$content</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  result &lt;- strsplit(result, <span class="cm-string">"\n"</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  result &lt;- result[[1]]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  result &lt;- result[grepl(<span class="cm-string">"Pfam"</span>, result)]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  result &lt;- strsplit(result,<span class="cm-string">'; '</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  result &lt;- lapply(result,function(x) x[2])</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  result &lt;- unlist(result)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  pfamAnnot[[i]] &lt;- unique(c(enGene,result))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  }</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">}</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">names(pfamAnnot) &lt;- names(results)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#pull ipr results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">ipr &lt;- read.delim(<span class="cm-string">'../../Genome_annotation/functionalAnnotation/HVAEP1.prot.longestIso.fa.tsv'</span>, header <span class="cm-operator">=</span> F)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">ipr<span class="cm-def">$V1</span> &lt;- gsub(<span class="cm-string">'_T(\\d+)[.]\\d+'</span>,<span class="cm-string">'_G\\1'</span>,ipr<span class="cm-def">$V1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#import TF IDs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">tfID &lt;- read.delim(<span class="cm-string">'../../Genome_annotation/functionalAnnotation/tfIDs.txt'</span><span class="cm-def">,header</span><span class="cm-operator">=</span>F)[,1]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#subset ipr table to include only TFs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">ipr.tf &lt;- ipr[ipr<span class="cm-def">$V1</span> %in% tfID,]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#get PF domains associated with all the TF entries in the ipr table</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">ipr.tf.pf &lt;- unique(ipr.tf[ipr.tf<span class="cm-def">$V4</span> <span class="cm-operator">==</span> <span class="cm-string">'Pfam'</span>,5])</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop pfam domains from swissprot that didn't show up in our ipr domain list from our hydra TFs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">pfamAnnot &lt;- lapply(pfamAnnot,function(x) x[x %in% ipr.tf.pf])</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#get the Hydra genes that contain the domains linked to each motif</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">pfamAnnot.genes &lt;- lapply(pfamAnnot,function(x) unique(ipr[ipr<span class="cm-def">$V5</span> %in% x,1]))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop motifs without a linked gene</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">pfamAnnot.genes &lt;- pfamAnnot.genes[sapply(pfamAnnot.genes,length) &gt; <span class="cm-number">0</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">saveRDS(pfamAnnot.genes,<span class="cm-string">'motifBindPfam.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 2046px;"></div><div class="CodeMirror-gutters" style="display: none; height: 2046px;"></div></div></div></pre><h2 id='single-cell-motif-enrichment-analysis'><span>Single-Cell Motif Enrichment Analysis</span></h2><h3 id='analysis-using-bona-fide-binding-motifs'><span>Analysis Using Bona Fide Binding Motifs</span></h3><p><span>We performed the main motif enrichment and gene co-expression regulator predictions within a single script (</span><code>hydraRegulators.R</code><span>). We have opted to break this script into smaller code blocks in this document to facilitate explanation.</span></p><h4 id='performing-gene-set-enrichment-analyses-1'><span>Performing Gene Set Enrichment Analyses</span></h4><p><span>After the initial steps of setting up the environment for the analysis, our first step was to generate gene sets. These gene sets, which group genes by the motifs present in their presumptive regulatory regions, served as the basis for a gene set enrichment analysis (GSEA; described in the next section). Essentially, the GSEA analysis determined if the presence of a particular binding motif in a gene&#39;s regulatory sequence was positively correlated with the gene having a higher NMF gene score for a particular metagene (the metagene gene score reflects how well a gene&#39;s expression pattern mimics the metagene expression pattern).</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulators.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">Seurat</span>)</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">tidyverse</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">rstudioapi</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">glmGamPoi</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">plotly</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">RColorBrewer</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">patchwork</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">plyr</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">RColorBrewer</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">gplots</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">viridis</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">fgsea</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#utility to convert transcript ID to gene ID</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">t2g</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">vapply</span>(<span class="cm-variable">x</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">y</span>) <span class="cm-variable">gsub</span>(<span class="cm-string">'HVAEP1_T(</span><span class="cm-string-2">\\</span><span class="cm-string">d+)[.]</span><span class="cm-string-2">\\</span><span class="cm-string">d'</span>,<span class="cm-string">'HVAEP1-G</span><span class="cm-string-2">\\</span><span class="cm-string">1'</span>,<span class="cm-variable">y</span>),<span class="cm-string">""</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">}</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">setwd</span>(<span class="cm-variable">dirname</span>(<span class="cm-variable">getActiveDocumentContext</span>()<span class="cm-operator cm-dollar">$</span><span class="cm-variable">path</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####prep gene sets and gene scores for gsea####</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load NMF gene Z scores</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.delim</span>(<span class="cm-string">'final/whole_unfilt_fine_narrow.gene_spectra_score.k_56.dt_0_13.txt'</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#reformat gene names because cnmf got rid of the underscores</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">colnames</span>(<span class="cm-variable">gscores</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gsub</span>(<span class="cm-string">'[.]'</span>,<span class="cm-string">'_'</span>,<span class="cm-variable">colnames</span>(<span class="cm-variable">gscores</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load annotated conserved transcription factor binding sites</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mots</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.delim</span>(<span class="cm-string">'../../alignment_conservation/motifAnnot/conMotATAC_finalhits.txt'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop motifs that weren't near a gene</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mots</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mots</span>[<span class="cm-variable">complete.cases</span>(<span class="cm-variable">mots</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop all genes that have no conserved motifs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gscores</span>[,<span class="cm-variable">colnames</span>(<span class="cm-variable">gscores</span>) <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">mots</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">gene_id</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#create a gene set for each binding motif</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mots</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">split</span>(<span class="cm-variable">mots</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">gene_id</span>,<span class="cm-variable">mots</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">peak_id</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop duplicated gene IDs in each gene set</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mots</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">mots</span>,<span class="cm-variable">unique</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 924px;"></div><div class="CodeMirror-gutters" style="display: none; height: 924px;"></div></div></div></pre><p><span>To perform the motif enrichment analysis we used the </span><code>fgseaMultilevel</code><span> function from the </span><code>fgsea</code><span> package. This function requires a list of gene sets and a named vector of scores. For our analysis, the gene sets list was a list made up of vectors of gene name, with the name for each vector being the motif ID associated with the gene IDs contained in that vector. The named vector of scores correspond to the gene scores for a specific metagene. </span></p><p><span>For our analysis, we performed gsea iteratively for each of the 56 metagenes we identified in the </span><em><span>Hydra</span></em><span> single-cell atlas. We then dropped (i.e., converted to 0) any enrichment score that failed to pass our significance cutoff (adjusted p-value ≤ 0.01). We then combined all the enrichment results into a single data frame.</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulators.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####ID enriched motifs####</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#go through the gene scores of each metagene and use them as weights</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#to order genes for a gsea analysis</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#genes are linked to motifs though the motif gene set we made above</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">set.seed</span>(<span class="cm-number">12345</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichList</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">gscores</span>,<span class="cm-number">1</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">gscores.use</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">x</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">names</span>(<span class="cm-variable">gscores.use</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">colnames</span>(<span class="cm-variable">gscores</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">fgseaMultilevel</span>(<span class="cm-variable">mots</span>, <span class="cm-variable">gscores.use</span>, <span class="cm-variable">minSize</span> <span class="cm-operator">=</span> <span class="cm-number">25</span>, <span class="cm-variable">nproc</span> <span class="cm-operator">=</span> <span class="cm-number">6</span>,<span class="cm-variable">scoreType</span> <span class="cm-operator">=</span> <span class="cm-string">'pos'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#Set all normalized enrichment scores (NES) to zero if</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#they fail to pass the significance cutoff</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#then subset the results table to include just the </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#motif name and enrichment score</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichList</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">enrichList</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">x</span>[<span class="cm-variable">x</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">padj</span> <span class="cm-operator">&gt;</span> <span class="cm-number">0.01</span>,<span class="cm-string">'NES'</span>] <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-number">0</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">x</span>[,<span class="cm-variable">c</span>(<span class="cm-number">1</span>,<span class="cm-number">6</span>)]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#Fix issue where enrichment returned NAs (just set to zero)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichList</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">enrichList</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">x</span>[<span class="cm-variable">is.na</span>(<span class="cm-variable">x</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">NES</span>),]<span class="cm-operator cm-dollar">$</span><span class="cm-variable">NES</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-number">0</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-builtin">return</span>(<span class="cm-variable">x</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#save motif names column from output to use for rownames later</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rowN</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">enrichList</span>[[<span class="cm-number">1</span>]][,<span class="cm-number">1</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#extract just the enrichment scores for each metagene</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichList</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">enrichList</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">x</span>[,<span class="cm-number">2</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#combine enrichment scores into an enrichment matrix spanning all metagenes</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichDF</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">as.data.frame</span>(<span class="cm-variable">do.call</span>(<span class="cm-variable">cbind</span>,<span class="cm-variable">enrichList</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rownames</span>(<span class="cm-variable">enrichDF</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">rowN</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">pathway</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">colnames</span>(<span class="cm-variable">enrichDF</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">paste0</span>(<span class="cm-string">'mg'</span>,<span class="cm-number">1</span><span class="cm-operator">:</span><span class="cm-variable">ncol</span>(<span class="cm-variable">enrichDF</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#save results to save time later</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">saveRDS</span>(<span class="cm-variable">enrichDF</span>,<span class="cm-string">'enrichDFnCDSATAC.rds'</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 968px;"></div><div class="CodeMirror-gutters" style="display: none; height: 968px;"></div></div></div></pre><h4 id='generating-single-cell-motif-enrichment-scores-1'><span>Generating Single-Cell Motif Enrichment Scores</span></h4><p><span>To map our enrichment results onto our Seurat object (for visualization and to correlate the results with gene expression data), we used the NMF cell scores. NMF cell scores are weights that specify how strongly each metagene contributes to a cell&#39;s overall transcriptional profile. We therefore used these cell scores to generate a weighted average of enrichment scores for each motif at a single-cell level, such that the enrichment score from a high scoring metagene contributed more strongly to a cell&#39;s enrichment score than a lowly scoring metagene.</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulators.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####Propagate enrichment scores to seurat object####</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#import NMF metagene cell scores</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">k56.usage</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.delim</span>(<span class="cm-string">'final/whole_unfilt_fine_narrow.usages.k_56.dt_0_13.consensus.txt'</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#normalize cell scores so that that all the scores for a single cell sum to 1</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">k56.usage</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">t</span>(<span class="cm-variable">apply</span>(<span class="cm-variable">k56.usage</span>,<span class="cm-number">1</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">x</span><span class="cm-operator">/</span><span class="cm-variable">sum</span>(<span class="cm-variable">x</span>)))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motScores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">k56.usage</span>, <span class="cm-number">1</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">apply</span>(<span class="cm-variable">enrichDF</span>,<span class="cm-number">1</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">y</span>) <span class="cm-variable">sum</span>(<span class="cm-variable">x</span><span class="cm-operator">*</span><span class="cm-variable">y</span>), <span class="cm-variable">simplify</span> <span class="cm-operator">=</span> <span class="cm-variable">T</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motScores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">as.data.frame</span>(<span class="cm-variable">t</span>(<span class="cm-variable">motScores</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motScores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">motScores</span>[,<span class="cm-variable">colSums</span>(<span class="cm-variable">motScores</span>) <span class="cm-operator">!=</span> <span class="cm-number">0</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load doublet-free, annotated atlas Seurat object</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ds</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">readRDS</span>(<span class="cm-string">'../dropSeqMapping/nonDubLabeledSeurat.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motScores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">motScores</span>[<span class="cm-variable">rownames</span>(<span class="cm-variable">ds</span>@<span class="cm-variable">meta.data</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">saveRDS</span>(<span class="cm-variable">motScores</span>,<span class="cm-string">'motScores.rds'</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 440px;"></div><div class="CodeMirror-gutters" style="display: none; height: 440px;"></div></div></div></pre><p><span>We then used the enrichment scores we calculated for our Hydra atlas to generate a heat map of single-cell enrichment scores averaged across different cell types. For this plot, and for other downstream analyses, we dropped any motifs that were not found to show signs of conservation in our cross-species whole-genome alignment (specified in the </span><code>motifConservationStats.csv</code><span> file generated in </span><code>07_genomeConservation.md</code><span>). This section of code also makes use of the file </span><code>k56_mg_annot.csv</code><span>, which was manually generated and includes descriptive names for each metagene. There&#39;s also the file </span><code>mg_order.csv</code><span> which specifies the order in which to metagenes appear in the heat map. </span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulators.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####motif enrichment heatmap####</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load metagene descriptions</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mgD</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.csv</span>(<span class="cm-string">'k56_mg_annot.csv'</span>,<span class="cm-variable">header</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mgD</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V1</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">paste0</span>(<span class="cm-string">'mg'</span>,<span class="cm-variable">mgD</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#initialize plotting data object</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">enrichDF</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#rename the columns to have the more descriptive metagene name</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">colnames</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mgD</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V2</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#import statistics on motif conservation</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#we only want to look at motifs that showed</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#evidence of conservation</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motifCon</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.csv</span>(<span class="cm-string">'../../alignment_conservation/motifConservationStats.csv'</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motifCon.keep</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">rownames</span>(<span class="cm-variable">motifCon</span>[<span class="cm-variable">motifCon</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">res</span> <span class="cm-operator">==</span> <span class="cm-string">'enriched'</span>,])</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[<span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">motifCon.keep</span>,]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop an motifs not enriched in any clusters</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#and drop any clusters without any enriched motifs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[<span class="cm-variable">rowSums</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator">!=</span> <span class="cm-number">0</span>,]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[,<span class="cm-variable">colSums</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator">!=</span> <span class="cm-number">0</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#normalize scores by row</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#(each motif can have a max enrichment score of 1)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">t</span>(<span class="cm-variable">apply</span>(<span class="cm-variable">nesPlot</span>,<span class="cm-number">1</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">x</span><span class="cm-operator">/</span><span class="cm-variable">max</span>(<span class="cm-variable">x</span>)))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#bring in motif name/family information</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#used to make the rownames more readable/interpretable</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motInfo</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.csv</span>(<span class="cm-string">'../../alignment_conservation/motifDB/motifInfo.csv'</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot.rnames</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mapvalues</span>(<span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlot</span>),<span class="cm-variable">from</span><span class="cm-operator">=</span><span class="cm-variable">motInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">ID</span>,<span class="cm-variable">to</span><span class="cm-operator">=</span><span class="cm-variable">motInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">name</span>,<span class="cm-variable">warn_missing</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot.rnames</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">paste0</span>(<span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlot</span>),<span class="cm-string">'</span> <span class="cm-string">'</span>,<span class="cm-variable">nesPlot.rnames</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot.rnames</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#order metagenes to be grouped more logically</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mgOrder</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.csv</span>(<span class="cm-string">'mg_order.csv'</span>,<span class="cm-variable">header</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-atom">NULL</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mgOrder</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V2</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mgD</span>[<span class="cm-variable">mgOrder</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V1</span>,<span class="cm-string">'V2'</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mgOrder</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mgOrder</span>[<span class="cm-variable">mgOrder</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V2</span> <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">colnames</span>(<span class="cm-variable">nesPlot</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[,<span class="cm-variable">mgOrder</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V2</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#because one of the metagenes got dropped (batch effect mg)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#we have to go back and make sure we have no all 0 rows or columns</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[<span class="cm-variable">rowSums</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator">!=</span> <span class="cm-number">0</span>,]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[,<span class="cm-variable">colSums</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator">!=</span> <span class="cm-number">0</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#plot heatmap</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pdf</span>(<span class="cm-string">'motifHeatmap.pdf'</span>,<span class="cm-variable">width</span> <span class="cm-operator">=</span> <span class="cm-number">15</span>,<span class="cm-variable">height</span> <span class="cm-operator">=</span> <span class="cm-number">60</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">heatmap.2</span>(<span class="cm-variable">as.matrix</span>(<span class="cm-variable">nesPlot</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">Colv</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">scale</span> <span class="cm-operator">=</span> <span class="cm-string">'none'</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">dendrogram</span> <span class="cm-operator">=</span> <span class="cm-string">'none'</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">col</span> <span class="cm-operator">=</span> <span class="cm-variable">viridis</span>(<span class="cm-number">30</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">trace</span><span class="cm-operator">=</span><span class="cm-string">'none'</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">key</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">keysize</span> <span class="cm-operator">=</span> <span class="cm-number">0.1</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">margins</span> <span class="cm-operator">=</span> <span class="cm-variable">c</span>(<span class="cm-number">10</span>,<span class="cm-number">10</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">colsep</span> <span class="cm-operator">=</span> <span class="cm-variable">c</span>(<span class="cm-number">3</span>,<span class="cm-number">19</span>,<span class="cm-number">29</span>,<span class="cm-number">38</span>,<span class="cm-number">47</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">distfun</span> <span class="cm-operator">=</span> <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">as.dist</span>(<span class="cm-number">1</span><span class="cm-operator">-</span><span class="cm-variable">cor</span>(<span class="cm-variable">t</span>(<span class="cm-variable">x</span>))),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">hclustfun</span> <span class="cm-operator">=</span> <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">hclust</span>(<span class="cm-variable">x</span>, <span class="cm-variable">method</span><span class="cm-operator">=</span><span class="cm-string">"average"</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">dev.off</span>()</span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">write.csv</span>(<span class="cm-variable">nesPlot</span>,<span class="cm-string">'hydraEnrichmentMatrix.csv'</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 1540px;"></div><div class="CodeMirror-gutters" style="display: none; height: 1540px;"></div></div></div></pre><p><img src="resources/motifHeatmap.png" alt="motifHeat" style="zoom:10%;" /></p><h4 id='removing-redundant-motifs-from-the-enrichment-results'><span>Removing Redundant Motifs from the Enrichment Results</span></h4><p><span>This initial heat map had over 300 rows, and wasn&#39;t very compact. To reduce the number of rows in the plot, we removed redundant motif entries, which we defined as motifs with highly similar sequence composition and enrichment patterns. We had already grouped motifs according to sequence similarity. In the following section of the analysis, we performed another clustering analysis where we clustered motifs based on correlation scores calculated by comparing motif enrichment results. </span></p><p><span>We grouped the motifs in a way that integrated both the sequence-based and the enrichment-based clustering analyses, such that motifs were only considered redundant if they had similar sequence composition </span><strong><span>and</span></strong><span> enrichment results. To collapse a group of motifs that were flagged as redundant, we presented the averaged enrichment profile of all motifs within that group. </span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulators.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####reduced redundancy heatmap####</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#look at correlation in enrichment patterns across motifs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motEnCor</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">enrichDF</span>,<span class="cm-number">1</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">apply</span>(<span class="cm-variable">enrichDF</span>,<span class="cm-number">1</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">y</span>) <span class="cm-variable">cor</span>(<span class="cm-variable">x</span>,<span class="cm-variable">y</span>)))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motEnCor</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">motEnCor</span>[<span class="cm-operator">!</span><span class="cm-variable">is.na</span>(<span class="cm-variable">motEnCor</span>[,<span class="cm-number">1</span>]),<span class="cm-operator">!</span><span class="cm-variable">is.na</span>(<span class="cm-variable">motEnCor</span>[,<span class="cm-number">1</span>])]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#generate distance matrix from correlation coefficient</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">d</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">as.dist</span>(<span class="cm-number">1</span> <span class="cm-operator">-</span> <span class="cm-variable">motEnCor</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#use hierarchical clustering to group motifs based on correlation</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">hc1</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">hclust</span>(<span class="cm-variable">d</span>, <span class="cm-variable">method</span> <span class="cm-operator">=</span> <span class="cm-string">"average"</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pdf</span>(<span class="cm-string">'motDistClustEn.pdf'</span>,<span class="cm-variable">width</span><span class="cm-operator">=</span><span class="cm-number">150</span>,<span class="cm-variable">height</span><span class="cm-operator">=</span><span class="cm-number">20</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">plot</span>(<span class="cm-variable">hc1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">abline</span>(<span class="cm-variable">h</span> <span class="cm-operator">=</span> <span class="cm-number">0.4</span>, <span class="cm-variable">col</span> <span class="cm-operator">=</span> <span class="cm-string">'red'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">dev.off</span>()</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#cut the branches at 0.4 and group accordingly</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">clusts</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">cutree</span>(<span class="cm-variable">hc1</span>, <span class="cm-variable">h</span> <span class="cm-operator">=</span> <span class="cm-number">0.4</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">repMotifs</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">data.frame</span>(<span class="cm-variable">ID</span> <span class="cm-operator">=</span> <span class="cm-variable">names</span>(<span class="cm-variable">clusts</span>), <span class="cm-variable">clust</span> <span class="cm-operator">=</span> <span class="cm-variable">clusts</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load motif clustering (based on sequence composition alone)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motClust</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.csv</span>(<span class="cm-string">'../../alignment_conservation/motifDB/motif_clusters.csv'</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#extract just the JASPAR part of the motif ID</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motClust</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">jID</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gsub</span>(<span class="cm-string">'.*_'</span>,<span class="cm-string">''</span>,<span class="cm-variable">motClust</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">ID</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#subset to just focus on motifs still being considered for this analysis</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motClust</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">motClust</span>[<span class="cm-variable">motClust</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">jID</span> <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">repMotifs</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">ID</span>,]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#make row order identical between the two clustering results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motClust</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">motClust</span>[<span class="cm-variable">match</span>(<span class="cm-variable">repMotifs</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">ID</span>,<span class="cm-variable">motClust</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">jID</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#merge clustering results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motClust</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">enClust</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">repMotifs</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">clust</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#combine the two clustering results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#only consider motifs redundant if they</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#both clustering analyses grouped them together</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motClust</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">bClust</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">paste</span>(<span class="cm-variable">motClust</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">clust</span>,<span class="cm-variable">motClust</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">enClust</span>,<span class="cm-variable">sep</span><span class="cm-operator">=</span><span class="cm-string">'_'</span>)</span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">write.csv</span>(<span class="cm-variable">motClust</span>,<span class="cm-variable">file</span><span class="cm-operator">=</span><span class="cm-string">'finalMotClust.csv'</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 946px;"></div><div class="CodeMirror-gutters" style="display: none; height: 946px;"></div></div></div></pre><p><img src="resources/motDistClustEn.png" referrerpolicy="no-referrer" alt="motDistClustEn"></p><p><span>We then generated a new, &#39;low-redundancy&#39; version of the heat map. </span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulators.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#generating a reduced redundancy version of the heatmap</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlotLr</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">as.data.frame</span>(<span class="cm-variable">nesPlot</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#extract just jaspar ID</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlotLr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">mot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gsub</span>(<span class="cm-string">'</span> <span class="cm-string">.*'</span>,<span class="cm-string">''</span>,<span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlotLr</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#bring in clustering results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlotLr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">motClust</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">motClust</span>[<span class="cm-variable">match</span>(<span class="cm-variable">nesPlotLr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">mot</span>,<span class="cm-variable">motClust</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">jID</span>),<span class="cm-string">'bClust'</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#sum together enrichment scores across all metagenes for each motif</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlotLr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cumScore</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">nesPlotLr</span>[,<span class="cm-number">1</span><span class="cm-operator">:</span><span class="cm-number">53</span>],<span class="cm-number">1</span>,<span class="cm-variable">sum</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#select a representative motif from each cluster</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#base it on which motif had the strongest</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#overall enrichment signal (highest cumulative score)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#for the actuall heatmap, make the values an average</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#of all the motifs that were grouped into that cluster</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#first get the name of the representative motif</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlotLr</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlotLr</span>[<span class="cm-variable">order</span>(<span class="cm-variable">nesPlotLr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">motClust</span>,<span class="cm-operator">-</span><span class="cm-variable">nesPlotLr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cumScore</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlotLr.motNames</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlotLr</span>[<span class="cm-operator">!</span><span class="cm-variable">duplicated</span>(<span class="cm-variable">nesPlotLr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">motClust</span>),])</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#group motifs by cluster</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlotLr</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">split</span>(<span class="cm-variable">nesPlotLr</span>[<span class="cm-number">1</span><span class="cm-operator">:</span><span class="cm-number">53</span>],<span class="cm-variable">nesPlotLr</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">motClust</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#get average enrichment profile across all motifs with a cluster</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlotLr</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">nesPlotLr</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">apply</span>(<span class="cm-variable">x</span>,<span class="cm-number">2</span>,<span class="cm-variable">mean</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#collapse into new low redundancy heatmap matrix</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlotLr</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">do.call</span>(<span class="cm-variable">rbind</span>,<span class="cm-variable">nesPlotLr</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlotLr</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlotLr.motNames</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pdf</span>(<span class="cm-string">'motifHeatmapLR.pdf'</span>,<span class="cm-variable">width</span> <span class="cm-operator">=</span> <span class="cm-number">15</span>,<span class="cm-variable">height</span> <span class="cm-operator">=</span> <span class="cm-number">40</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">heatmap.2</span>(<span class="cm-variable">nesPlotLr</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">Colv</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">scale</span> <span class="cm-operator">=</span> <span class="cm-string">'none'</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">dendrogram</span> <span class="cm-operator">=</span> <span class="cm-string">'none'</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">col</span> <span class="cm-operator">=</span> <span class="cm-variable">viridis</span>(<span class="cm-number">30</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">trace</span><span class="cm-operator">=</span><span class="cm-string">'none'</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">key</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">keysize</span> <span class="cm-operator">=</span> <span class="cm-number">0.1</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">margins</span> <span class="cm-operator">=</span> <span class="cm-variable">c</span>(<span class="cm-number">10</span>,<span class="cm-number">10</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">colsep</span> <span class="cm-operator">=</span> <span class="cm-variable">c</span>(<span class="cm-number">3</span>,<span class="cm-number">19</span>,<span class="cm-number">29</span>,<span class="cm-number">38</span>,<span class="cm-number">47</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">distfun</span> <span class="cm-operator">=</span> <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">as.dist</span>(<span class="cm-number">1</span><span class="cm-operator">-</span><span class="cm-variable">cor</span>(<span class="cm-variable">t</span>(<span class="cm-variable">x</span>))),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">hclustfun</span> <span class="cm-operator">=</span> <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">hclust</span>(<span class="cm-variable">x</span>, <span class="cm-variable">method</span><span class="cm-operator">=</span><span class="cm-string">"average"</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">dev.off</span>()</span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">write.csv</span>(<span class="cm-variable">nesPlotLr</span>,<span class="cm-string">'hydraEnrichmentMatrixLR.csv'</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 1100px;"></div><div class="CodeMirror-gutters" style="display: none; height: 1100px;"></div></div></div></pre><p>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p><img src="resources/motifHeatmapLR.png" alt="motHeatLr" style="zoom:10%;" /></p><h4 id='identifying-candidate-regulators-of-enriched-motifs'><span>Identifying Candidate Regulators of Enriched Motifs</span></h4><p><span>We next linked enriched motifs to putative regulators. To do this we needed to correlate a motif&#39;s enrichment pattern in the single-cell atlas to the expression patterns of predicted transcription factors. One potential challenge for performing this type of correlation analysis involving single-cell gene expression data is that the low depth of scRNA-seq makes it susceptible to frequent &#39;drop-outs&#39;, where a gene with moderate expression has zero counts in a non-trivial number of cells. To mitigate the issues this might cause for the correlation analysis between motif enrichment and gene expression, we generated a matrix of imputed read counts based on our NMF analysis. </span></p><p><span>The goal of NMF is to generate two matrices (the cell score matrix and the gene score matrix) that, when multiplied together, create an approximation of the original data matrix (the single-cell gene expression matrix). We took advantage of this to create an expression matrix that roughly recapitulated the original expression data, but without any drop-outs, smoothing out the data and making it more suitable for correlation analyses. This was the matrix that we used to determine the correlation score between predicted transcription factors and enriched motifs.</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulators.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####Link TFs to Motifs####</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#look for putatively linked TFs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#based on pfam domain prediction and</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#expression profile</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#generate imputed gene expression based on NMF</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#makes correlation easier</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#reload gene scores</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">t</span>(<span class="cm-variable">read.delim</span>(<span class="cm-string">'final/whole_unfilt_fine_narrow.gene_spectra_tpm.k_56.dt_0_13.txt'</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rownames</span>(<span class="cm-variable">gscores</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gsub</span>(<span class="cm-string">'[.]'</span>,<span class="cm-string">'_'</span>,<span class="cm-variable">rownames</span>(<span class="cm-variable">gscores</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#multiply the gene and cell scores to create the NMF-derrived approximation for the gene</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#expression matrix</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">k56.usage</span>, <span class="cm-number">1</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">apply</span>(<span class="cm-variable">gscores</span>,<span class="cm-number">1</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">y</span>) <span class="cm-variable">sum</span>(<span class="cm-variable">x</span><span class="cm-operator">*</span><span class="cm-variable">y</span>), <span class="cm-variable">simplify</span> <span class="cm-operator">=</span> <span class="cm-variable">T</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">t</span>(<span class="cm-variable">gscores</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gscores</span>[<span class="cm-variable">rownames</span>(<span class="cm-variable">ds</span>@<span class="cm-variable">meta.data</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ds</span>@<span class="cm-variable">meta.data</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">nmfG</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gscores</span>[,<span class="cm-variable">which</span>(<span class="cm-variable">colnames</span>(<span class="cm-variable">gscores</span>) <span class="cm-operator">==</span> <span class="cm-string">'HVAEP1_G023106'</span>)]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">FeaturePlot</span>(<span class="cm-variable">ds</span>,<span class="cm-string">'nmfG'</span>,<span class="cm-variable">order</span> <span class="cm-operator">=</span> <span class="cm-variable">T</span>) <span class="cm-operator">+</span> <span class="cm-variable">NoLegend</span>() <span class="cm-operator">+</span> <span class="cm-variable">NoAxes</span>()</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">FeaturePlot</span>(<span class="cm-variable">ds</span>,<span class="cm-string">'HVAEP1-G023106'</span>,<span class="cm-variable">order</span> <span class="cm-operator">=</span> <span class="cm-variable">T</span>) <span class="cm-operator">+</span> <span class="cm-variable">NoLegend</span>() <span class="cm-operator">+</span> <span class="cm-variable">NoAxes</span>()</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">saveRDS</span>(<span class="cm-variable">gscores</span>,<span class="cm-string">'nmfNormExp.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">readRDS</span>(<span class="cm-string">'nmfNormExp.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#get list of genes linked to motifs by pfam domain</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">readRDS</span>(<span class="cm-string">'../../alignment_conservation/motifDB/motifBindPfam.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">pfamMot</span>[<span class="cm-variable">names</span>(<span class="cm-variable">pfamMot</span>) <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">gsub</span>(<span class="cm-string">'</span> <span class="cm-string">.*'</span>,<span class="cm-string">''</span>,<span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlot</span>))]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#convert to a dataframe</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">data.frame</span>(<span class="cm-variable">motID</span> <span class="cm-operator">=</span> <span class="cm-variable">unlist</span>(<span class="cm-variable">sapply</span>(<span class="cm-number">1</span><span class="cm-operator">:</span><span class="cm-variable">length</span>(<span class="cm-variable">pfamMot</span>), <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">rep</span>(<span class="cm-variable">names</span>(<span class="cm-variable">pfamMot</span>)[<span class="cm-variable">x</span>],<span class="cm-variable">length</span>(<span class="cm-variable">pfamMot</span>[[<span class="cm-variable">x</span>]])))),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">geneID</span> <span class="cm-operator">=</span> <span class="cm-variable">unlist</span>(<span class="cm-variable">pfamMot</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">geneID</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">t2g</span>(<span class="cm-variable">pfamMot</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">geneID</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#subset to genes present in drop-seq object</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">pfamMot</span>[<span class="cm-variable">gsub</span>(<span class="cm-string">'_'</span>,<span class="cm-string">'-'</span>,<span class="cm-variable">pfamMot</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">geneID</span>) <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">rownames</span>(<span class="cm-variable">ds</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#list of TFs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">tfList</span> <span class="cm-operator cm-arrow">&lt;-</span><span class="cm-variable">read.delim</span>(<span class="cm-string">'../../Genome_annotation/functionalAnnotation/tfIDs.txt'</span>)[,<span class="cm-number">1</span>,<span class="cm-variable">drop</span><span class="cm-operator">=</span><span class="cm-variable">T</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores.tf</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gscores</span>[,<span class="cm-variable">colnames</span>(<span class="cm-variable">gscores</span>) <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">tfList</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#check similarity between motif enrichment and gene expression</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corRes</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">gscores.tf</span>,<span class="cm-number">2</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">apply</span>(<span class="cm-variable">motScores</span>, <span class="cm-number">2</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">y</span>) <span class="cm-variable">cor</span>(<span class="cm-variable">x</span>,<span class="cm-variable">y</span>)))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">saveRDS</span>(<span class="cm-variable">corRes</span>,<span class="cm-string">'tfMotifCor.rds'</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 1188px;"></div><div class="CodeMirror-gutters" style="display: none; height: 1188px;"></div></div></div></pre><p><span>From this motif/transcription factor correlation matrix, we generated a summary table of all putative co-expression regulators. A candidate regulator was defined as a transcription factor that had a correlation score of 0.5 or greater with a motif that it could plausibly bind based on its Pfam domain content. This table included the gene ID of the candidate regulator as well as a list of all the enriched motifs that transcription factor could potentially be regulating. We also added some functional annotation data for the candidate regulator, such as predicted orthologs and Pfam domains.</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulators.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####Generate Summary Table####</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">pfamMot</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">pfamMot.res</span>[<span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">motID</span> <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">rownames</span>(<span class="cm-variable">corRes</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">pfamMot.res</span>[<span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">geneID</span> <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">colnames</span>(<span class="cm-variable">corRes</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#add in the correlation score calculated above</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cor</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">pfamMot.res</span>,<span class="cm-number">1</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">corRes</span>[<span class="cm-variable">x</span>[<span class="cm-number">1</span>],<span class="cm-variable">x</span>[<span class="cm-number">2</span>]]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">write.csv</span>(<span class="cm-variable">pfamMot.res</span>,<span class="cm-string">'pfamMotCor.csv'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#set a correlation score cutoff of 0.5</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">pfamMot.res</span>[<span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cor</span> <span class="cm-operator">&gt;=</span> <span class="cm-number">0.5</span>,]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#bring in motif name</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">motName</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mapvalues</span>(<span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">motID</span>,<span class="cm-variable">from</span><span class="cm-operator">=</span><span class="cm-variable">motInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">ID</span>,<span class="cm-variable">to</span><span class="cm-operator">=</span><span class="cm-variable">motInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">name</span>,<span class="cm-variable">warn_missing</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#collapse by gene to simplify</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">split</span>(<span class="cm-variable">pfamMot.res</span>,<span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">geneID</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#concatenate correlated motifs into a single string, ordered to have most correlated motifs first</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">pfamMot.res</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) {</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">newDf</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">x</span>[<span class="cm-variable">order</span>(<span class="cm-operator">-</span><span class="cm-variable">x</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cor</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">newDf</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">comboMot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">paste0</span>(<span class="cm-variable">newDf</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">motID</span>,<span class="cm-string">'</span> <span class="cm-string">('</span>,<span class="cm-variable">newDf</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">motName</span>,<span class="cm-string">'):'</span>,<span class="cm-variable">round</span>(<span class="cm-variable">newDf</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cor</span>,<span class="cm-variable">digits</span> <span class="cm-operator">=</span> <span class="cm-number">3</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">newDf</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">data.frame</span>(<span class="cm-variable">geneID</span> <span class="cm-operator">=</span> <span class="cm-variable">newDf</span>[<span class="cm-number">1</span>,<span class="cm-string">'geneID'</span>], <span class="cm-variable">corMots</span> <span class="cm-operator">=</span> <span class="cm-variable">paste</span>(<span class="cm-variable">newDf</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">comboMot</span>,<span class="cm-variable">collapse</span> <span class="cm-operator">=</span> <span class="cm-string">'; '</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-builtin">return</span>(<span class="cm-variable">newDf</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">do.call</span>(<span class="cm-variable">rbind</span>,<span class="cm-variable">pfamMot.res</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#bring in gene annotations (orthology, domains, blast hits)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gInfo</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.csv</span>(<span class="cm-string">'../../Orthofinder/HVAEP1_annotation.csv'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">ortho</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mapvalues</span>(<span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">geneID</span>,<span class="cm-variable">from</span><span class="cm-operator">=</span><span class="cm-variable">gsub</span>(<span class="cm-string">'HVAEP1_T(</span><span class="cm-string-2">\\</span><span class="cm-string">d+)[.].*'</span>,<span class="cm-string">'HVAEP1_G</span><span class="cm-string-2">\\</span><span class="cm-string">1'</span>,<span class="cm-variable">gInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">H_vulgarisAEP</span>),<span class="cm-variable">to</span> <span class="cm-operator">=</span> <span class="cm-variable">gInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">EnsemblLongName</span>, <span class="cm-variable">warn_missing</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">pfam</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mapvalues</span>(<span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">geneID</span>,<span class="cm-variable">from</span><span class="cm-operator">=</span><span class="cm-variable">gsub</span>(<span class="cm-string">'HVAEP1_T(</span><span class="cm-string-2">\\</span><span class="cm-string">d+)[.].*'</span>,<span class="cm-string">'HVAEP1_G</span><span class="cm-string-2">\\</span><span class="cm-string">1'</span>,<span class="cm-variable">gInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">H_vulgarisAEP</span>),<span class="cm-variable">to</span> <span class="cm-operator">=</span> <span class="cm-variable">gInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">PFAM_NAME</span>, <span class="cm-variable">warn_missing</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">gb</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mapvalues</span>(<span class="cm-variable">pfamMot.res</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">geneID</span>,<span class="cm-variable">from</span><span class="cm-operator">=</span><span class="cm-variable">gsub</span>(<span class="cm-string">'HVAEP1_T(</span><span class="cm-string-2">\\</span><span class="cm-string">d+)[.].*'</span>,<span class="cm-string">'HVAEP1_G</span><span class="cm-string-2">\\</span><span class="cm-string">1'</span>,<span class="cm-variable">gInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">H_vulgarisAEP</span>),<span class="cm-variable">to</span> <span class="cm-operator">=</span> <span class="cm-variable">gInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">genBankAnnotation</span>, <span class="cm-variable">warn_missing</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">write.csv</span>(<span class="cm-variable">pfamMot.res</span>,<span class="cm-string">'regulatorResults.csv'</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 990px;"></div><div class="CodeMirror-gutters" style="display: none; height: 990px;"></div></div></div></pre><p><span>To generate the motif enrichment and gene expression plots provided in the paper we used the template below. The motif IDs and gene IDs were swapped to create the other plots presented throughout the text.</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulators.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####Plot correlated motifs/TFs####</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">unlink</span>(<span class="cm-string">'motOlapPlots'</span>,<span class="cm-variable">recursive</span><span class="cm-operator">=</span><span class="cm-variable">T</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">dir.create</span>(<span class="cm-string">'motOlapPlots'</span>,<span class="cm-variable">showWarnings</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#ebf motif</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ds</span>@<span class="cm-variable">meta.data</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">testMot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">motScores</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable-3">`MA1637.1`</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">FeaturePlot</span>(<span class="cm-variable">ds</span>,<span class="cm-string">'testMot'</span>,<span class="cm-variable">order</span> <span class="cm-operator">=</span> <span class="cm-variable">T</span>,<span class="cm-variable">pt.size</span> <span class="cm-operator">=</span> <span class="cm-number">0.8</span>) <span class="cm-operator">+</span> <span class="cm-variable">NoAxes</span>()</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ggsave</span>(<span class="cm-string">'motOlapPlots/ebfM.png'</span>,<span class="cm-variable">width</span><span class="cm-operator">=</span><span class="cm-number">9</span>,<span class="cm-variable">height</span><span class="cm-operator">=</span><span class="cm-number">8</span>,<span class="cm-variable">dpi</span><span class="cm-operator">=</span><span class="cm-number">300</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#ebf</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">FeaturePlot</span>(<span class="cm-variable">ds</span>,<span class="cm-variable">t2g</span>(<span class="cm-string">'HVAEP1_T005780.2'</span>),<span class="cm-variable">order</span> <span class="cm-operator">=</span> <span class="cm-variable">T</span>,<span class="cm-variable">pt.size</span> <span class="cm-operator">=</span> <span class="cm-number">0.8</span>) <span class="cm-operator">+</span> <span class="cm-variable">NoAxes</span>()</span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ggsave</span>(<span class="cm-string">'motOlapPlots/ebf.png'</span>,<span class="cm-variable">width</span><span class="cm-operator">=</span><span class="cm-number">9</span>,<span class="cm-variable">height</span><span class="cm-operator">=</span><span class="cm-number">8</span>,<span class="cm-variable">dpi</span><span class="cm-operator">=</span><span class="cm-number">300</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 264px;"></div><div class="CodeMirror-gutters" style="display: none; height: 264px;"></div></div></div></pre><p><img src="resources/ebfM.png" referrerpolicy="no-referrer" alt="ebfM"></p><p><img src="resources/ebf.png" referrerpolicy="no-referrer" alt="ebfGexp"></p><h3 id='analysis-using-shuffled-binding-motifs'><span>Analysis Using Shuffled Binding Motifs</span></h3><p><span>We next wanted to determine the extent to which the enrichment patterns we observed above were due to true biological signal, as opposed to being artifacts driven by chance. To test this, we repeated the single-cell motif enrichment analysis using shuffled instead of bona fide transcription factor binding motifs. We had already generated a list of shuffled motifs for the analysis described in </span><code>07_genomeConservation.md</code><span>. In addition we had subjected predicted instances of these shuffled motifs to the same filtering criteria (i.e., requiring conservation across multiple </span><em><span>Hydra</span></em><span> genomes and localization to open chromatin) we used for generating binding site predictions for our database of bona fide binding sites. </span></p><p><span>Because the shuffled motifs should be random, non-functional sequences, any enrichment patterns we identify using the shuffled motifs will likely be artifacts. If we get similar enrichment results using both the shuffled and bona fide versions of a motif, we can determine that our enrichment results are likely not biologically meaningful; however, if the bona fide motif has a dramatically different enrichment pattern than the shuffled motif, then we can conclude that the enrichment results are not purely the result of sequence bias.</span></p><h4 id='performing-gene-set-enrichment-analyses-2'><span>Performing Gene Set Enrichment Analyses</span></h4><p><span>To repeat our enrichment analysis using shuffled motifs, we used the </span><code>conShufMotsATAC_finalhits.txt</code><span> table to generate gene sets. We then performed gsea for each metagene in the </span><em><span>Hydra</span></em><span> single cell atlas using these shuffled motif gene sets:</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulatorsShuf.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">Seurat</span>)</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">tidyverse</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">rstudioapi</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">glmGamPoi</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">plotly</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">RColorBrewer</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">patchwork</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">plyr</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">RColorBrewer</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">gplots</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">viridis</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">library</span>(<span class="cm-variable">fgsea</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#utility to convert transcript ID to gene ID</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">t2g</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">vapply</span>(<span class="cm-variable">x</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">y</span>) <span class="cm-variable">gsub</span>(<span class="cm-string">'HVAEP1_T(</span><span class="cm-string-2">\\</span><span class="cm-string">d+)[.]</span><span class="cm-string-2">\\</span><span class="cm-string">d'</span>,<span class="cm-string">'HVAEP1-G</span><span class="cm-string-2">\\</span><span class="cm-string">1'</span>,<span class="cm-variable">y</span>),<span class="cm-string">""</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">}</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">setwd</span>(<span class="cm-variable">dirname</span>(<span class="cm-variable">getActiveDocumentContext</span>()<span class="cm-operator cm-dollar">$</span><span class="cm-variable">path</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####prep gene sets and gene scores for gsea####</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load NMF gene Z scores</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.delim</span>(<span class="cm-string">'final/whole_unfilt_fine_narrow.gene_spectra_score.k_56.dt_0_13.txt'</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#reformat gene names because cnmf got rid of the underscores</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">colnames</span>(<span class="cm-variable">gscores</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gsub</span>(<span class="cm-string">'[.]'</span>,<span class="cm-string">'_'</span>,<span class="cm-variable">colnames</span>(<span class="cm-variable">gscores</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load annotated conserved transcription factor binding sites</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mots</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.delim</span>(<span class="cm-string">'../../alignment_conservation/motifAnnot/conShufMotsATAC_finalhits.txt'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop motifs that weren't near a gene</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mots</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mots</span>[<span class="cm-variable">complete.cases</span>(<span class="cm-variable">mots</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop all genes that have no conserved motifs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">gscores</span>[,<span class="cm-variable">colnames</span>(<span class="cm-variable">gscores</span>) <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">mots</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">gene_id</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#create a gene set for each binding motif</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mots</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">split</span>(<span class="cm-variable">mots</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">gene_id</span>,<span class="cm-variable">mots</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">peak_id</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop duplicated gene IDs in each gene set</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mots</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">mots</span>,<span class="cm-variable">unique</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####ID enriched motifs####</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#go through the gene scores of each metagene and use them as weights</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#to order genes for a gsea analysis</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#genes are linked to motifs though the motif gene set we made above</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">set.seed</span>(<span class="cm-number">12345</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichList</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">gscores</span>,<span class="cm-number">1</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">gscores.use</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">x</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">names</span>(<span class="cm-variable">gscores.use</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">colnames</span>(<span class="cm-variable">gscores</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">fgseaMultilevel</span>(<span class="cm-variable">mots</span>, <span class="cm-variable">gscores.use</span>, <span class="cm-variable">minSize</span> <span class="cm-operator">=</span> <span class="cm-number">25</span>, <span class="cm-variable">nproc</span> <span class="cm-operator">=</span> <span class="cm-number">6</span>,<span class="cm-variable">scoreType</span> <span class="cm-operator">=</span> <span class="cm-string">'pos'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#Set all normalized enrichment scores (NES) to zero if</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#they fail to pass the significance cutoff</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#then subset the results table to include just the </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#motif name and enrichment score</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichList</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">enrichList</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">x</span>[<span class="cm-variable">x</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">padj</span> <span class="cm-operator">&gt;</span> <span class="cm-number">0.01</span>,<span class="cm-string">'NES'</span>] <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-number">0</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">x</span>[,<span class="cm-variable">c</span>(<span class="cm-number">1</span>,<span class="cm-number">6</span>)]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#Fix issue where enrichment returned NAs (just set to zero)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichList</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">enrichList</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">x</span>[<span class="cm-variable">is.na</span>(<span class="cm-variable">x</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">NES</span>),]<span class="cm-operator cm-dollar">$</span><span class="cm-variable">NES</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-number">0</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-builtin">return</span>(<span class="cm-variable">x</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#save motif names column from output to use for rownames later</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rowN</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">enrichList</span>[[<span class="cm-number">1</span>]][,<span class="cm-number">1</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#extract just the enrichment scores for each metagene</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichList</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">enrichList</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-variable">x</span>[,<span class="cm-number">2</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#combine enrichment scores into an enrichment matrix spanning all metagenes</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichDF</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">as.data.frame</span>(<span class="cm-variable">do.call</span>(<span class="cm-variable">cbind</span>,<span class="cm-variable">enrichList</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rownames</span>(<span class="cm-variable">enrichDF</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">rowN</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">pathway</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">colnames</span>(<span class="cm-variable">enrichDF</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">paste0</span>(<span class="cm-string">'mg'</span>,<span class="cm-number">1</span><span class="cm-operator">:</span><span class="cm-variable">ncol</span>(<span class="cm-variable">enrichDF</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#save results to save time later</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">saveRDS</span>(<span class="cm-variable">enrichDF</span>,<span class="cm-string">'enrichDFnCDSATAC_shuf.rds'</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 1914px;"></div><div class="CodeMirror-gutters" style="display: none; height: 1914px;"></div></div></div></pre><h4 id='generating-single-cell-motif-enrichment-scores-2'><span>Generating Single-Cell Motif Enrichment Scores</span></h4><p><span>We then used NMF cell scores to translate the metagene enrichment results into single-cell enrichment scores for each motif:</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulatorsShuf.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####Propagate enrichment scores to seurat object####</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#import NMF metagene cell scores</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">k56.usage</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.delim</span>(<span class="cm-string">'final/whole_unfilt_fine_narrow.usages.k_56.dt_0_13.consensus.txt'</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#normalize cell scores so that that all the scores for a single cell sum to 1</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">k56.usage</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">t</span>(<span class="cm-variable">apply</span>(<span class="cm-variable">k56.usage</span>,<span class="cm-number">1</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">x</span><span class="cm-operator">/</span><span class="cm-variable">sum</span>(<span class="cm-variable">x</span>)))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motScores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">k56.usage</span>, <span class="cm-number">1</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">apply</span>(<span class="cm-variable">enrichDF</span>,<span class="cm-number">1</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">y</span>) <span class="cm-variable">sum</span>(<span class="cm-variable">x</span><span class="cm-operator">*</span><span class="cm-variable">y</span>), <span class="cm-variable">simplify</span> <span class="cm-operator">=</span> <span class="cm-variable">T</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motScores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">as.data.frame</span>(<span class="cm-variable">t</span>(<span class="cm-variable">motScores</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motScores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">motScores</span>[,<span class="cm-variable">colSums</span>(<span class="cm-variable">motScores</span>) <span class="cm-operator">!=</span> <span class="cm-number">0</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load doublet-free, annotated atlas Seurat object</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ds</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">readRDS</span>(<span class="cm-string">'../dropSeqMapping/nonDubLabeledSeurat.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motScores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">motScores</span>[<span class="cm-variable">rownames</span>(<span class="cm-variable">ds</span>@<span class="cm-variable">meta.data</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">saveRDS</span>(<span class="cm-variable">motScores</span>,<span class="cm-string">'motScores_shuf.rds'</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 440px;"></div><div class="CodeMirror-gutters" style="display: none; height: 440px;"></div></div></div></pre><p><span>We then visualized these motif enrichment scores using a motif by cell type heat map:</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulatorsShuf.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####motif enrichment heatmap####</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load metagene descriptions</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mgD</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.csv</span>(<span class="cm-string">'k56_mg_annot.csv'</span>,<span class="cm-variable">header</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mgD</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V1</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">paste0</span>(<span class="cm-string">'mg'</span>,<span class="cm-variable">mgD</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#initialize plotting data object</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">enrichDF</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#rename the columns to have the more descriptive metagene name</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">colnames</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mgD</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V2</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop an motifs not enriched in any clusters</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#and drop any clusters without any enriched motifs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[<span class="cm-variable">rowSums</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator">!=</span> <span class="cm-number">0</span>,]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[,<span class="cm-variable">colSums</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator">!=</span> <span class="cm-number">0</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#normalize scores by row</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#(each motif can have a max enrichment score of 1)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">t</span>(<span class="cm-variable">apply</span>(<span class="cm-variable">nesPlot</span>,<span class="cm-number">1</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">x</span><span class="cm-operator">/</span><span class="cm-variable">max</span>(<span class="cm-variable">x</span>)))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#bring in motif name/family information</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#used to make the rownames more readable/interpretable</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">motInfo</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.csv</span>(<span class="cm-string">'../../alignment_conservation/motifDB/motifInfo.csv'</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-number">1</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot.rnames</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mapvalues</span>(<span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlot</span>),<span class="cm-variable">from</span><span class="cm-operator">=</span><span class="cm-variable">motInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">ID</span>,<span class="cm-variable">to</span><span class="cm-operator">=</span><span class="cm-variable">motInfo</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">name</span>,<span class="cm-variable">warn_missing</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot.rnames</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">paste0</span>(<span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlot</span>),<span class="cm-string">'</span> <span class="cm-string">'</span>,<span class="cm-variable">nesPlot.rnames</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rownames</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot.rnames</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#order metagenes to be grouped more logically</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mgOrder</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">read.csv</span>(<span class="cm-string">'mg_order.csv'</span>,<span class="cm-variable">header</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>,<span class="cm-variable">row.names</span> <span class="cm-operator">=</span> <span class="cm-atom">NULL</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mgOrder</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V2</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mgD</span>[<span class="cm-variable">mgOrder</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V1</span>,<span class="cm-string">'V2'</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">mgOrder</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">mgOrder</span>[<span class="cm-variable">mgOrder</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V2</span> <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">colnames</span>(<span class="cm-variable">nesPlot</span>),]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[,<span class="cm-variable">mgOrder</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">V2</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#because one of the metagenes got dropped (batch effect mg)</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#we have to go back and make sure we have no all 0 rows or columns</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[<span class="cm-variable">rowSums</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator">!=</span> <span class="cm-number">0</span>,]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">nesPlot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">nesPlot</span>[,<span class="cm-variable">colSums</span>(<span class="cm-variable">nesPlot</span>) <span class="cm-operator">!=</span> <span class="cm-number">0</span>]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#plot heatmap</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pdf</span>(<span class="cm-string">'motifHeatmap_shuf.pdf'</span>,<span class="cm-variable">width</span> <span class="cm-operator">=</span> <span class="cm-number">10</span>,<span class="cm-variable">height</span> <span class="cm-operator">=</span> <span class="cm-number">60</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">heatmap.2</span>(<span class="cm-variable">as.matrix</span>(<span class="cm-variable">nesPlot</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">Colv</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">scale</span> <span class="cm-operator">=</span> <span class="cm-string">'none'</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">dendrogram</span> <span class="cm-operator">=</span> <span class="cm-string">'none'</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">col</span> <span class="cm-operator">=</span> <span class="cm-variable">viridis</span>(<span class="cm-number">30</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">trace</span><span class="cm-operator">=</span><span class="cm-string">'none'</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">key</span> <span class="cm-operator">=</span> <span class="cm-variable">F</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">keysize</span> <span class="cm-operator">=</span> <span class="cm-number">0.1</span>,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">margins</span> <span class="cm-operator">=</span> <span class="cm-variable">c</span>(<span class="cm-number">10</span>,<span class="cm-number">10</span>),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">distfun</span> <span class="cm-operator">=</span> <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">as.dist</span>(<span class="cm-number">1</span><span class="cm-operator">-</span><span class="cm-variable">cor</span>(<span class="cm-variable">t</span>(<span class="cm-variable">x</span>))),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">hclustfun</span> <span class="cm-operator">=</span> <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">hclust</span>(<span class="cm-variable">x</span>, <span class="cm-variable">method</span><span class="cm-operator">=</span><span class="cm-string">"average"</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">dev.off</span>()</span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">write.csv</span>(<span class="cm-variable">nesPlot</span>,<span class="cm-string">'hydraEnrichmentMatrix_shuf.csv'</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 1320px;"></div><div class="CodeMirror-gutters" style="display: none; height: 1320px;"></div></div></div></pre><p><img src="resources/motifHeatmap_shuf.png" alt="motifHeatmap_shuf" style="zoom:10%;" /></p><p><span>This heat map revealed a large number of enriched motifs when using the shuffled motif set, although markedly fewer metagenes had enriched motifs when compared to the bona fide motif set. </span></p><h3 id='comparing-enrichment-results-for-genuine-and-shuffled-motifs'><span>Comparing Enrichment Results for Genuine and Shuffled Motifs</span></h3><p><span>To determine if the enrichment patterns we observed were similar for both the shuffled and bona fide versions of each motif (indicating that the enrichment patterns were driven by sequence bias), we calculated correlation scores for the enrichment results for the bona fide and shuffled motif results. We found that the enrichment results were highly different for the vast majority of motifs, indicating that the enrichment patterns we observed were not primarily being driven by sequence bias artifacts.</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulatorsShuf.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">####compare shuffled results to the results from bona fide motifs####</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#first just compare raw enrichment results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#import the bona fide enrichment results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichDF.bf</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">readRDS</span>(<span class="cm-string">'enrichDFnCDSATAC.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#need to make sure the rows are identical between the two dfs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichDF.bf.comp</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">lapply</span>(<span class="cm-variable">rownames</span>(<span class="cm-variable">enrichDF</span>),<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-keyword">if</span>(<span class="cm-variable">x</span> <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">rownames</span>(<span class="cm-variable">enrichDF.bf</span>)){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;<span class="cm-builtin">return</span>(<span class="cm-variable">enrichDF.bf</span>[<span class="cm-variable">x</span>,])</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  } <span class="cm-keyword">else</span> {</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;<span class="cm-builtin">return</span>(<span class="cm-variable">rep</span>(<span class="cm-number">0</span>,<span class="cm-variable">ncol</span>(<span class="cm-variable">enrichDF.bf</span>)))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  }</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">enrichDF.bf.comp</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">do.call</span>(<span class="cm-variable">rbind</span>,<span class="cm-variable">enrichDF.bf.comp</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">rownames</span>(<span class="cm-variable">enrichDF.bf.comp</span>) <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">rownames</span>(<span class="cm-variable">enrichDF</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#measure similarity between the shuffled and bona fide motif enrichment results</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corRes</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">vapply</span>(<span class="cm-number">1</span><span class="cm-operator">:</span><span class="cm-variable">nrow</span>(<span class="cm-variable">enrichDF</span>), <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">cor</span>(<span class="cm-variable">as.numeric</span>(<span class="cm-variable">enrichDF</span>[<span class="cm-variable">x</span>,]),<span class="cm-variable">as.numeric</span>(<span class="cm-variable">enrichDF.bf.comp</span>[<span class="cm-variable">x</span>,])),<span class="cm-variable">numeric</span>(<span class="cm-number">1</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corRes</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">data.frame</span>(<span class="cm-variable">mot</span> <span class="cm-operator">=</span> <span class="cm-variable">rownames</span>(<span class="cm-variable">enrichDF</span>),<span class="cm-variable">score</span><span class="cm-operator">=</span><span class="cm-variable">corRes</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#need to fix cases where the cor score couldn't be calculated because of invariance </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#in one of the two DFs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corRes</span>[<span class="cm-variable">is.na</span>(<span class="cm-variable">corRes</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">score</span>),<span class="cm-string">'score'</span>] <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-number">0</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">summary</span>(<span class="cm-variable">corRes</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">score</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 682px;"></div><div class="CodeMirror-gutters" style="display: none; height: 682px;"></div></div></div></pre><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang=""><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  Min.  1st Qu. &nbsp; Median &nbsp; &nbsp; Mean  3rd Qu. &nbsp; &nbsp; Max. </span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">-0.10616  0.00000  0.00000  0.03438  0.00000  1.00000 </span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 44px;"></div><div class="CodeMirror-gutters" style="display: none; height: 44px;"></div></div></div></pre><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ggplot</span>(<span class="cm-variable">corRes</span>,<span class="cm-variable">aes</span>(<span class="cm-variable">x</span><span class="cm-operator">=</span><span class="cm-string">'motifCor'</span>,<span class="cm-variable">y</span><span class="cm-operator">=</span><span class="cm-variable">score</span>)) <span class="cm-operator">+</span> <span class="cm-variable">geom_violin</span>() <span class="cm-operator">+</span> <span class="cm-variable">geom_jitter</span>(<span class="cm-variable">width</span> <span class="cm-operator">=</span> <span class="cm-number">0.3</span>, <span class="cm-variable">height</span> <span class="cm-operator">=</span> <span class="cm-number">0.02</span>) <span class="cm-operator">+</span> <span class="cm-variable">theme_bw</span>()</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 22px;"></div><div class="CodeMirror-gutters" style="display: none; height: 22px;"></div></div></div></pre><p><img src="resources/enCorScor.png" referrerpolicy="no-referrer" alt="enCorScor"></p><p><span>One of the criteria we used for identifying transcriptional regulators was that their transcriptional expression pattern should overlap with the enrichment pattern of their target motif. We therefore sought to determine if the enrichment patterns we calculated using bona fide motifs tended to correlate more closely with the expression patterns of their candidate regulators than the enrichment patterns calculated using shuffled motifs. </span></p><p><span>To do this we calculated a correlation score for each possible gene/motif pair (determined above based on pfam domains) by comparing the NMF-imputed gene expression values for the gene and the corresponding motif enrichment scores. We calculated this correlation score for both the bona fide and shuffled versions of the motif. We then compared the two correlation scores (one for bona fide and one for shuffled) using a scatter plot, including only on those gene/motif pairs that had a high correlation score (above 0.4) for either motif version.</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulatorsShuf.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#next check to see if the protein/motif correspondence is better with the non-shuffled motif than it is with the</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#shuffled motif</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load imputed gene expression matrix</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">gscores</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">readRDS</span>(<span class="cm-string">'nmfNormExp.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#get list of genes linked to motifs by pfam domain</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">pfamMot</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">readRDS</span>(<span class="cm-string">'../../alignment_conservation/motifDB/motifBindPfam.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#check similarity between motif enrichment and gene expression</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corRes</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">gscores.tf</span>,<span class="cm-number">2</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">apply</span>(<span class="cm-variable">motScores</span>, <span class="cm-number">2</span>, <span class="cm-keyword">function</span>(<span class="cm-variable">y</span>) <span class="cm-variable">cor</span>(<span class="cm-variable">x</span>,<span class="cm-variable">y</span>)))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">saveRDS</span>(<span class="cm-variable">corRes</span>,<span class="cm-string">'tfMotifCor_shuf.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corRes</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">readRDS</span>(<span class="cm-string">'tfMotifCor_shuf.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#load gene-motif correlation scores for bona fide motifs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corRes.bf</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">readRDS</span>(<span class="cm-string">'tfMotifCor.rds'</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#generate table of possible motif/gene pairs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corComp</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">data.frame</span>(<span class="cm-variable">motID</span> <span class="cm-operator">=</span> <span class="cm-variable">unlist</span>(<span class="cm-variable">sapply</span>(<span class="cm-number">1</span><span class="cm-operator">:</span><span class="cm-variable">length</span>(<span class="cm-variable">pfamMot</span>), <span class="cm-keyword">function</span>(<span class="cm-variable">x</span>) <span class="cm-variable">rep</span>(<span class="cm-variable">names</span>(<span class="cm-variable">pfamMot</span>)[<span class="cm-variable">x</span>],<span class="cm-variable">length</span>(<span class="cm-variable">pfamMot</span>[[<span class="cm-variable">x</span>]])))),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">geneID</span> <span class="cm-operator">=</span> <span class="cm-variable">unlist</span>(<span class="cm-variable">pfamMot</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#pull correlation score for each gene/motif pair for the shuffled motif data</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cor</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">corComp</span>,<span class="cm-number">1</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-keyword">if</span>(<span class="cm-variable">x</span>[<span class="cm-number">1</span>] <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">rownames</span>(<span class="cm-variable">corRes</span>) <span class="cm-operator">&amp;</span> <span class="cm-variable">x</span>[<span class="cm-number">2</span>] <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">colnames</span>(<span class="cm-variable">corRes</span>)){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;<span class="cm-builtin">return</span>(<span class="cm-variable">corRes</span>[<span class="cm-variable">x</span>[<span class="cm-number">1</span>],<span class="cm-variable">x</span>[<span class="cm-number">2</span>]])</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  } <span class="cm-keyword">else</span> {</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;<span class="cm-builtin">return</span>(<span class="cm-number">0</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  }</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#pull correlation score for each gene/motif pair for the bona fide motif data</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">corBF</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">apply</span>(<span class="cm-variable">corComp</span>,<span class="cm-number">1</span>,<span class="cm-keyword">function</span>(<span class="cm-variable">x</span>){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;<span class="cm-keyword">if</span>(<span class="cm-variable">x</span>[<span class="cm-number">1</span>] <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">rownames</span>(<span class="cm-variable">corRes.bf</span>) <span class="cm-operator">&amp;</span> <span class="cm-variable">x</span>[<span class="cm-number">2</span>] <span class="cm-operator cm-variable-2">%in%</span> <span class="cm-variable">colnames</span>(<span class="cm-variable">corRes.bf</span>)){</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;<span class="cm-builtin">return</span>(<span class="cm-variable">corRes.bf</span>[<span class="cm-variable">x</span>[<span class="cm-number">1</span>],<span class="cm-variable">x</span>[<span class="cm-number">2</span>]])</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  } <span class="cm-keyword">else</span> {</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp;<span class="cm-builtin">return</span>(<span class="cm-number">0</span>)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">  }</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">})</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#drop gene/motif pairs that had zeroes for both bona fide and</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#shuffled motifs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corComp</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">corComp</span>[<span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cor</span> <span class="cm-operator">!=</span> <span class="cm-number">0</span> <span class="cm-operator">|</span> <span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">corBF</span> <span class="cm-operator">!=</span> <span class="cm-number">0</span>,]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#for plot, drop gene/motif pairs that did not high scores for either </span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#shuffled or bona fide motif sequences</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corComp.p</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">corComp</span>[<span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cor</span> <span class="cm-operator">&gt;</span> <span class="cm-number">0.4</span> <span class="cm-operator">|</span> <span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">corBF</span> <span class="cm-operator">&gt;</span> <span class="cm-number">0.4</span>,]</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#visualize correspondence between shuffled and bona fide motifs</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ggplot</span>(<span class="cm-variable">corComp.p</span>,<span class="cm-variable">aes</span>(<span class="cm-variable">x</span><span class="cm-operator">=</span><span class="cm-variable">cor</span>,<span class="cm-variable">y</span><span class="cm-operator">=</span><span class="cm-variable">corBF</span>)) <span class="cm-operator">+</span> <span class="cm-variable">geom_point</span>() <span class="cm-operator">+</span> <span class="cm-variable">theme_bw</span>()</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 1166px;"></div><div class="CodeMirror-gutters" style="display: none; height: 1166px;"></div></div></div></pre><p><img src="resources/motEnCorComp.png" referrerpolicy="no-referrer" alt="motEnCorComp"></p><p><span>(Note: the x-axis corresponds to scores calculated using shuffled motif enrichment patterns and the y-axis corresponds to scores calculated using bona fide motif enrichment patterns)</span></p><p><span>The plot above shows that for any given gene/motif pair, the degree of correspondence between enrichment and expression for the shuffled motif has virtually no correlation to the degree of correspondence for the bona fide motif. This can also be demonstrated more concisely by just calculating the correlation between the two scores:</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">cor</span>(<span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cor</span>,<span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">corBF</span>)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 22px;"></div><div class="CodeMirror-gutters" style="display: none; height: 22px;"></div></div></div></pre><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang=""><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">[1] 0.09411901</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 22px;"></div><div class="CodeMirror-gutters" style="display: none; height: 22px;"></div></div></div></pre><p><span>In addition, the plot reveals a trend where unshuffled motifs tend to have higher correlation scores than the shuffled motifs, as there are more points with high values on the y-axis than the x-axis. This can also be demonstrated using a box plot showing the distribution of correlation scores for bona fide and shuffled motifs:</span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulatorsShuf.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#is the correspondence between motif enrichment and gene expression</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#stronger for bona fide motifs than for shuffled motifs?</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">corComp.vln</span> <span class="cm-operator cm-arrow">&lt;-</span> <span class="cm-variable">data.frame</span>(<span class="cm-variable">corType</span><span class="cm-operator">=</span><span class="cm-variable">rep</span>(<span class="cm-variable">c</span>(<span class="cm-string">'shuf'</span>,<span class="cm-string">'bf'</span>),<span class="cm-variable">c</span>(<span class="cm-variable">nrow</span>(<span class="cm-variable">corComp</span>),<span class="cm-variable">nrow</span>(<span class="cm-variable">corComp</span>))),</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;<span class="cm-variable">corScore</span><span class="cm-operator">=</span><span class="cm-variable">c</span>(<span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cor</span>,<span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">corBF</span>))</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">ggplot</span>(<span class="cm-variable">corComp.vln</span>,<span class="cm-variable">aes</span>(<span class="cm-variable">x</span><span class="cm-operator">=</span><span class="cm-variable">corType</span>,<span class="cm-variable">y</span><span class="cm-operator">=</span><span class="cm-variable">corScore</span>)) <span class="cm-operator">+</span> <span class="cm-variable">geom_boxplot</span>() <span class="cm-operator">+</span> <span class="cm-variable">theme_bw</span>()</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 132px;"></div><div class="CodeMirror-gutters" style="display: none; height: 132px;"></div></div></div></pre><p><img src="resources/geneMotCorComp.png" referrerpolicy="no-referrer" alt="geneMotCorComp"></p><p><span>Although most gene/motif pairs showed no correlation (even for bona fide motif sequences) the overall correspondence between motif enrichment and gene expression was significantly higher for bona fide motifs than for shuffled motifs. </span></p><p><span>(</span><em><span>snippet from 02_enrichmentAnalysis/hydraRegulatorsShuf.R</span></em><span>)</span></p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="R"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang="r"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#is the difference significant?</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-variable">t.test</span>(<span class="cm-variable">x</span><span class="cm-operator">=</span><span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">cor</span>,<span class="cm-variable">y</span><span class="cm-operator">=</span><span class="cm-variable">corComp</span><span class="cm-operator cm-dollar">$</span><span class="cm-variable">corBF</span>)</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 44px;"></div><div class="CodeMirror-gutters" style="display: none; height: 44px;"></div></div></div></pre><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang=""><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span>Welch Two Sample t-test</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="" cm-zwsp="">
</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">data:  corComp$cor and corComp$corBF</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">t = -21.551, df = 36786, p-value &lt; 2.2e-16</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">alternative hypothesis: true difference in means is not equal to 0</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">95 percent confidence interval:</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> -0.03683734 -0.03069529</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">sample estimates:</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; mean of x &nbsp;  mean of y </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">-0.008893753  0.024872563 </span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 220px;"></div><div class="CodeMirror-gutters" style="display: none; height: 220px;"></div></div></div></pre><p><span>Overall, these results strongly suggest that the enrichment patterns we observed using our bona fide motif set are not purely driven by sequence bias artifacts. In addition, the enrichment patterns of unshuffled motifs more closely matched the expression patterns of their putative regulators than would be expected by chance.</span></p><p>&nbsp;</p><h2 id='files-associated-with-this-document'><span>Files Associated with This Document</span></h2><p>&nbsp;</p><pre class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" lang="" style="break-inside: unset;"><div class="CodeMirror cm-s-inner cm-s-null-scroll CodeMirror-wrap" lang=""><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 9px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">10_hydraRegulators/</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">├── 01_motifPrep</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── clusterMots.R</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>R script that uses a similarity matrix generated by homerMotCompare.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>to perform hierarchical clustering of the motifs in pooledJasparNR.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>based on sequence composition. </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── conMotATACAnnot.json</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>JSON config file used by the UROPA pipeline to link predicted </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>transcription factor binding sites to their target genes based on </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>proximity.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── conMotATAC_finalhits.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Output generated by UROPA linking predicted transcription factor</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>binding sites to their target genes based on proximity.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── conShufMotATACAnnot.json</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>JSON config file used by the UROPA pipeline to link conserved instances </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>of shuffled transcription factor binding motifs to their target genes </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>based on proximity.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── conShufMotsATAC_finalhits.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Output generated by UROPA linking conserved instances of shuffled </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>transcription factor binding motifs to their target genes based on </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>proximity.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── fullTF.csv</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Full interproscan protein domain annotations for all predicted </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>transcription factors in the AEP assembly gene models.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── homerMotCompare.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Shell script that uses the compareMotifs.pl script from HOMER to </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>generate a similarity matrix comparing the sequence composition</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>of all motifs in pooledJasparNR.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── jasparIDs.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Text file that lists all JASPAR motif IDs in pooledJasparNR.txt. Used</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>by protDBLink.R.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── motComp.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Text file containing a similarity matrix comparing the sequence </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>composition of all motifs in pooledJasparNR.txt. Generated by </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>homerMotCompare.sh</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── motifBindPfam.rds</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Binary R object file containing a list object that links individual</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>JASPAR motifs to AEP gene models that could potential bind those </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>motifs based on pfam protein domain annotations.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── motif_clusters.csv</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Table containing the clustering results from clusterMots.R. Motifs</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp;  were grouped using a hierarchical clustering analysis based on</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp; &nbsp; &nbsp;  sequence similarities.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── motifInfo.csv</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Table that provides the name and transcription factor family associated</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>with each motif in pooledJasparNR.txt.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── pooledJasparNR.homer.motifs</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>HOMER-formatted version of the pooledJasparNR.txt motif database.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── protDBLink.R</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>R script that links AEP gene models to putative target binding motifs</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>based on pfam domain annotations. Specifically, it identifies the</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>DNA binding domain associated with each JASPAR motif and then links</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>that motif to all AEP gene models that also contain that binding</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>domain.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; ├── tfIDs.txt</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Text file listing the gene ID of all putative transcription factors</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>in the AEP gene models.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">│&nbsp;&nbsp; └── tfList.R</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>R script that uses inteproscan protein domain annotations to identify</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>putative transcription factors in the AEP gene models. It does this</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>using a combination of GO terms and lists of known DNA binding domains.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">└── 02_enrichmentAnalysis</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── enrichDFnCDSATAC.rds</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Binary R object file containing a motif by metagene matrix of normalized</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>enrichment scores (NES) for the Hydra single cell atlas. Higher values</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>indicate stronger enrichment. NES values that did not pass our significance</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>theshold (P. adj &lt; 0.01) were set to zero.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── enrichDFnCDSATAC_shuf.rds</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Binary R object file containing a motif by metagene matrix of normalized</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>enrichment scores (NES) for the Hydra single cell atlas calculated using </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>shuffled control versions of each motif. Higher values indicate stronger </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>enrichment. NES values that did not pass our significance theshold (P. </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>adj &lt; 0.01) were set to zero.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── finalMotClust.csv</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Table containing the final motif clustering results that grouped JASPAR</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>motifs based on both sequence composition (results taken from </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>motif_clusters.csv) and motif enrichment patterns. This was used to reduce</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>the redundancy of the final enrichment results matrix </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>(hydraEnrichmentMatrixLR.csv)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── hydraEnrichmentMatrix.csv</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Motif by cell type matrix of enrichment scores in the Hydra single cell</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>atlas. These values were used to generate the heatmaps presented in the</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>paper. </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── hydraEnrichmentMatrixLR.csv</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Low redundancy version of hydraEnrichmentMatrix.csv where rows (i.e., motifs)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>have been collapsed according to the clustering in finalMotClust.csv.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── hydraEnrichmentMatrix_shuf.csv</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Motif by cell type matrix of enrichment scores in the Hydra single cell</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>atlas calculated using shuffled control versions of each motif.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── hydraRegulators.R</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>R script that uses gene set enrichment analysis to identify motifs enriched</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>in each metagene in the Hydra single cell atlas. It then maps these values</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>onto individual cells to generate single-cell enrichment scores. Finally,</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>it compares motif enrichment patterns to the expression patterns of putative</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>transcription factors that could plausibly bind each motif to identify</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>condidate regulators of cell-type-specific expression.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── hydraRegulatorsShuf.R</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>R script that repeats the enrichment analysis from hydraRegulators.R, but </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>instead uses shuffled control versions of each motif. It then compares</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>the enrichment results for the shuffled and non-shuffled versions to </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>evaluate the degree to which the bona fide motif enrichment results</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>are meaningfully different from the results generated using random noise</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>(simulated using the shuffled motifs).</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── k56_mg_annot.csv</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Table that provides descriptive names for each metagene in the Hydra single</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>cell atlas.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── mg_order.csv</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Table that specifies the column order of the motif enrichment heatmaps </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>(e.g., hydraEnrichmentMatrixLR.csv)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── motScores.rds</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Binary R object file containing a motif by cell matrix of motif enrichment </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>scores for the Hydra single cell atlas.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── motScores_shuf.rds</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Binary R object file containing a motif by cell matrix of motif enrichment </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>scores for the Hydra single cell atlas calculated using shuffled control </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>versions of each motif.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── nmfNormExp.rds</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Binary R object file containing a gene by cell matrix of imputed/smoothened</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>expression values for the Hydra single cell atlas calculated using NMF. Used</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>to facilitate single cell gene expression correlation analyses.</span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── pfamMotCor.csv</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Table containing correlation scores for every motif/gene pair specified in </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>motifBindPfam.rds. The score reflects how well the enrichment pattern for the</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>motif matched the expression pattern of the putative regulator.</span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── regulatorResults.csv</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Table listing all candidate regulators of cell type specific transcription</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>in the Hydra single cell atlas. All putative transcription factors whose </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>expression pattern was correlated (correlation score threshold of ≥ 0.5)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>with the enrichment pattern of at least one motif were included in the table.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>The table also lists all motifs that passed the correlation threshold for each</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>transcription factor, as well as some functional annotation information for each </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>candidate regulator.</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  ├── tfMotifCor.rds</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Motif-by-gene correlation matrix comparing single-cell GSEA enrichment scores to </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>the NMF-imputed expression patterns of putative transcription factors in the </span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"> &nbsp;  <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Hydra atlas.</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>└── tfMotifCor_shuf.rds</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>Motif-by-gene correlation matrix comparing single-cell GSEA enrichment scores of </span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>shuffled control motifs to the NMF-imputed expression patterns of putative </span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span>transcription factors in the Hydra atlas.</span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" class="cm-tab-wrap-hack" style="padding-right: 0.1px;"><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-tab" role="presentation" cm-text="	">  </span></span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom-width: 0px; border-bottom-style: solid; border-bottom-color: transparent; top: 3102px;"></div><div class="CodeMirror-gutters" style="display: none; height: 3102px;"></div></div></div></pre><p>&nbsp;</p><p>&nbsp;</p></div></div>
</body>
</html>