# New *Hydra* genomes reveal conserved principles of hydrozoan transcriptional regulation

This repository contains step-by-step descriptions of all analyses and associated code related to the following manuscript:

>Cazet JF, Siebert S, Morris Little H,  Bertemes P, Primack AS, Ladurner P, Achrainer M, Fredriksen MT, Moreland RT, Singh S, Zhang S, Wolfsberg TG, Schnitzler CE, Baxevanis AD,  Simakov O, Hobmayer B, Juliano CE. A chromosome-scale epigenetic map of  the *Hydra* genome reveals conserved regulators of cell state.  Genome Res. 2023 Jan 13:gr.277040.122. doi: [10.1101/gr.277040.122](https://doi.org/10.1101/gr.277040.122). PMID: [36639202](https://pubmed.ncbi.nlm.nih.gov/36639202/).

The manuscript is also accompanied by a genome portal, available [here](https://research.nhgri.nih.gov/HydraAEP/), that allows users to interact with and download the data generated in this study. A BLAST server is available to search for genes of interest in the *H. oligactis* and strain AEP *H. vulgaris* gene models. The portal includes an interactive genome browser for visualizing gene models, repetitive regions, ATAC-seq and CUT&Tag peaks, ATAC-seq and CUT&Tag read density, and sequence conservation across the AEP assembly. The website also features an interactive ShinyCell portal for viewing the AEP-aligned *Hydra* single-cell atlas. 

**Important note: the naming conventions for gene models and assembly scaffolds on the genome portal are different from the conventions used throughout this repository. Please refer to the README within the `ID_Conversion` subdirectory for additional information and resources that can be used to interconvert between the two naming conventions**

### Structure and intent of this repository

This repository is organized around markdown documents that are each focused on one particular computational aspect of the manuscript. Each markdown includes all code used for the analysis in question along with accompanying text that explains the code's purpose and rationale. Each markdown is also accompanied by a folder that contains the original script files used for the analysis as well as files generated by the analysis itself. Descriptions for all files within each folder can be found at the bottom of the accompanying markdown document.

Our intention in generating this repository was to document the methodology we used to produce the results reported in the manuscript in sufficient detail for other researchers to reproduce our findings. However, the code is written in a manner that relies on directory/file structures and software path configurations that are specific to the systems on which the analyses were initially performed. This original file organization is not recapitulated by this repository. In addition, because of file size limitations, we are not able to provide all necessary files for every analysis via GitHub. As such, users will need to modify the paths within each script and download additional files from other sources (described below) for the code to run properly after the repository has been cloned.

### Accessing additional necessary files

All files necessary for performing the analyses described in this repository are available through the [*Hydra vulgaris*, strain AEP genome portal](https://research.nhgri.nih.gov/HydraAEP/download/index.cgi?dl=fa). Specifically, we provide complete versions of the folders that accompany each markdown document in this repository, including all files that were too large to host on GitHub. We also provide all sequencing data as well as R binary files containing various versions of the AEP-mapped single-cell RNA-seq atlas formatted as Seurat objects (v4).

Raw sequencing data is also available via NCBI under the BioProject ID PRJNA816482. The strain AEP *H. vulgaris* genome assembly is hosted on GenBank under the accession JALDPZ010000000 and the *H. oligactis* assembly is hosted under the accession JALDAD010000000. 
